{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SLC25A46 - Genotyping-imputed GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "GP2 genotyping imputed data: Release 7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Set paths\n",
    "- Installing packages\n",
    "\n",
    "### Making a working directory\n",
    "\n",
    "### Copy Over Files\n",
    "\n",
    "### Create a covariate file with GP2 data\n",
    "\n",
    "### Annotation \n",
    "\n",
    "- Turn binary files into VCF\n",
    "- Annotate using ANNOVAR\n",
    "\n",
    "### Case/Control Analysis\n",
    "\n",
    "- All variants (Association and Logistic regression)\n",
    "- Coding (Association and Logistic regression)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"chrom://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name @ `WORKSPACE_NAME`: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project @ `BILLING_PROJECT_ID`: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data @ `WORKSPACE_BUCKET`: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'path/'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "AMP_RELEASE_GATK_PATH = os.path.join(AMP_RELEASE_PATH, 'gatk')\n",
    "AMP_WGS_RELEASE_PATH = 'path/'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')\n",
    "\n",
    "## GP2 v7.0\n",
    "## Explicitly define release v7.0 path \n",
    "GP2_RELEASE_PATH = 'path/'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "print('GP2 v7.0')\n",
    "print(f'Path to GP2 v7.0 Clinical Data @ `GP2_CLINICAL_RELEASE_PATH`: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Metadata @ `GP2_META_RELEASE_PATH`: {GP2_META_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Raw Genotype Data @ `GP2_RAW_GENO_PATH`: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 Imputed Genotype Data @ `GP2_IMPUTED_GENO_PATH`: {GP2_IMPUTED_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 summary statistics: {GP2_SUMSTAT_RELEASE_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/plink2_linux_x86_64_20240504.zip\n",
    "unzip -o plink2_linux_x86_64_20240504.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_x86_64_20240504.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_x86_64_20240504.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Making a working directory"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Can use any ancestry - using EAS just as example"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry = 'EAS'\n",
    "WORK_DIR = f'SLC25A46_{ancestry}'\n",
    "\n",
    "! mkdir {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy Over Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check directory where GP2 Tier 2 data is\n",
    "print(\"List available imputed genotype information in GP2 (broken down by ancestry)\")\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##  Imputed genotype GP2 Tier 2 data\n",
    "# SLC25A46 coordinates: Chromosome chr5:110,738,136-110,765,161 https://www.genecards.org/cgi-bin/carddisp.pl?gene=SLC25A46&keywords=SLC25A46\n",
    "ancestry = 'EAS'  # CHANGE THE CHROMOSOME DEPENDING ON THE GENE AND THE ANCESTRY\n",
    "WORK_DIR = f'SLC25A46_{ancestry}'\n",
    "\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_IMPUTED_GENO_PATH}/{ancestry}/chr5_{ancestry}_release7.* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## clinical data \n",
    "\n",
    "## master key\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_CLINICAL_RELEASE_PATH}/master_key_release7_final.csv {WORK_DIR}')\n",
    "\n",
    "## related file \n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_META_RELEASE_PATH}/related_samples/{ancestry}_release7.related {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## PCs\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RAW_GENO_PATH}/{ancestry}/{ancestry}_release7.eigenvec {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5_EAS_release7.log\tchr5_EAS_release7.pvar\tmaster_key_release7_final.csv\r\n",
      "chr5_EAS_release7.pgen\tEAS_release7.eigenvec\r\n",
      "chr5_EAS_release7.psam\tEAS_release7.related\r\n"
     ]
    }
   ],
   "source": [
    "## Check files\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the master key\n",
    "key = pd.read_csv(f'{WORK_DIR}/master_key_release7_final.csv')\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subsetting to keep only a few columns \n",
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype_for_qc', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'label']]\n",
    "# Renaming the columns\n",
    "key.rename(columns = {'GP2sampleID':'IID',\n",
    "                                     'baseline_GP2_phenotype_for_qc':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Subset to keep ancestry of interest \n",
    "ancestry_key = key[key['label']==ancestry].copy()\n",
    "ancestry_key.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load information about related individuals in the ancestry analyzed\n",
    "related_df = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.related')\n",
    "print(related_df.shape)\n",
    "related_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of just one set of related people\n",
    "related_list = list(related_df['IID1'])\n",
    "\n",
    "# Check value counts of related and remove only one related individual\n",
    "ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "\n",
    "# Check size\n",
    "print(ancestry_key.shape)\n",
    "ancestry_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert phenotype to binary (1/2)\n",
    "## Assign conditions so case=2 and controls=1, and -9 otherwise (matching PLINK convention)\n",
    "    # PD = 2; control = 1\n",
    "pheno_mapping = {\"PD\": 2, \"Control\": 1}\n",
    "ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')\n",
    "ancestry_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO\n",
       "2       2657\n",
       "1       2572\n",
       "<NA>      44\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of pheno\n",
    "ancestry_key['PHENO'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Get the PCs\n",
    "pcs = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.eigenvec', sep='\\t')\n",
    "pcs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Split the single column into multiple columns\n",
    "#pcs_split = pcs['#FID,IID,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10'].str.split(',', expand=True)\n",
    "\n",
    "selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "\n",
    "# Drop the first row (since it's now the column names)\n",
    "pcs = pcs.drop(0)\n",
    "\n",
    "# Reset the index to remove any potential issues\n",
    "pcs = pcs.reset_index(drop=True)\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "print(pcs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check value counts of related and remove only one related individual\n",
    "# pcs = pcs[~pcs[\"IID\"].isin(related_list)]\n",
    "\n",
    "# Check size\n",
    "print(pcs.shape)\n",
    "pcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SEX\n",
       "Male                          3388\n",
       "Female                        1879\n",
       "Other/Unknown/Not Reported       6\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of SEX     \n",
    "ancestry_key['SEX'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert phenotype to binary (1/2)\n",
    "## Assign conditions so female=2 and men=1, and -9 otherwise (matching PLINK convention)\n",
    "    # Female = 2; Male = 1\n",
    "sex_mapping = {\"Female\": 2, \"Male\": 1}\n",
    "ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')\n",
    "ancestry_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SEX\n",
       "1       3388\n",
       "2       1879\n",
       "<NA>       6\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of SEX     \n",
    "ancestry_key['SEX'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
       "       'AAO', 'label', 'PHENO'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Make covariate file\n",
    "df = pd.merge(pcs, ancestry_key, on='IID', how='left')\n",
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['FID'] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Clean up and keep columns we need \n",
    "final_df = df[['FID','IID', 'SEX', 'AGE', 'PHENO','PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df['PHENO'] = final_df['PHENO'].fillna(-9)\n",
    "final_df['AGE'] = final_df['AGE'].fillna(-9)\n",
    "final_df['SEX'] = final_df['SEX'].fillna(-9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SEX\n",
       "1     3127\n",
       "2     1805\n",
       "-9     206\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Update sex to plink format: 1=male, 2=female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df[(final_df['PHENO']==2)&(final_df['AGE']==-9)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df[(final_df['PHENO']==1)&(final_df['AGE']==-9)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make file of sample IDs to keep \n",
    "samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "samples_toKeep.to_csv(f'{WORK_DIR}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make your covariate file\n",
    "final_df.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5_EAS_release7.log\tchr5_EAS_release7.pvar\tEAS_release7.related\r\n",
      "chr5_EAS_release7.pgen\tEAS_covariate_file.txt\tEAS.samplestoKeep\r\n",
      "chr5_EAS_release7.psam\tEAS_release7.eigenvec\tmaster_key_release7_final.csv\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_covariate_file.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry}_covariate_file.txt')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}.samplestoKeep {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry}.samplestoKeep')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Extract the region using PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_EAS/EAS_SLC25A46.log.\n",
      "Options in effect:\n",
      "  --chr 5\n",
      "  --from-bp 110738136\n",
      "  --make-bed\n",
      "  --out SLC25A46_EAS/EAS_SLC25A46\n",
      "  --pfile SLC25A46_EAS/chr5_EAS_release7\n",
      "  --to-bp 110765161\n",
      "\n",
      "Start time: Thu May 16 02:57:54 2024\n",
      "3676 MiB RAM detected, ~1456 available; reserving 1392 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "5139 samples (1870 females, 3269 males; 5139 founders) loaded from\n",
      "SLC25A46_EAS/chr5_EAS_release7.psam.\n",
      "11444438 variants loaded from SLC25A46_EAS/chr5_EAS_release7.pvar.\n",
      "1 binary phenotype loaded (2646 cases, 2453 controls).\n",
      "1951 variants remaining after main filters.\n",
      "Writing SLC25A46_EAS/EAS_SLC25A46.fam ... done.\n",
      "Writing SLC25A46_EAS/EAS_SLC25A46.bim ... done.\n",
      "Writing SLC25A46_EAS/EAS_SLC25A46.bed ... done.\n",
      "End time: Thu May 16 02:58:05 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "WORK_DIR = f'SLC25A46_{ancestry}'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {WORK_DIR}/chr5_{ancestry}_release7 \\\n",
    "--chr 5 \\\n",
    "--from-bp 110738136 \\\n",
    "--to-bp 110765161 \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5\tchr5:110738152:C:T\t0\t110738152\tT\tC\r\n",
      "5\tchr5:110738155:C:G\t0\t110738155\tG\tC\r\n",
      "5\tchr5:110738163:A:G\t0\t110738163\tG\tA\r\n",
      "5\tchr5:110738172:G:A\t0\t110738172\tA\tG\r\n",
      "5\tchr5:110738251:G:A\t0\t110738251\tA\tG\r\n",
      "5\tchr5:110738257:G:C\t0\t110738257\tC\tG\r\n",
      "5\tchr5:110738260:G:A\t0\t110738260\tA\tG\r\n",
      "5\tchr5:110738267:A:T\t0\t110738267\tT\tA\r\n",
      "5\tchr5:110738270:A:G\t0\t110738270\tG\tA\r\n",
      "5\tchr5:110738286:CA:C\t0\t110738286\tC\tCA\r\n"
     ]
    }
   ],
   "source": [
    "# Visualize bim file\n",
    "! head {WORK_DIR}/{ancestry}_SLC25A46.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Visualize fam file\n",
    "! head {WORK_DIR}/{ancestry}_SLC25A46.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Visualize bed file\n",
    "#! head {WORK_DIR}/{ancestry}_SLC25A46.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_EAS/EAS_SLC25A46.log.\n",
      "Options in effect:\n",
      "  --bfile SLC25A46_EAS/EAS_SLC25A46\n",
      "  --export vcf id-paste=iid\n",
      "  --mac 2\n",
      "  --out SLC25A46_EAS/EAS_SLC25A46\n",
      "\n",
      "Start time: Thu May 16 02:58:12 2024\n",
      "3676 MiB RAM detected, ~1540 available; reserving 1476 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "5139 samples (1870 females, 3269 males; 5139 founders) loaded from\n",
      "SLC25A46_EAS/EAS_SLC25A46.fam.\n",
      "1951 variants loaded from SLC25A46_EAS/EAS_SLC25A46.bim.\n",
      "1 binary phenotype loaded (2646 cases, 2453 controls).\n",
      "Calculating allele frequencies... done.\n",
      "1742 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "209 variants remaining after main filters.\n",
      "--export vcf to SLC25A46_EAS/EAS_SLC25A46.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu May 16 02:58:12 2024\n"
     ]
    }
   ],
   "source": [
    "## Turn binary files into VCF\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_SLC25A46.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_SLC25A46.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 SLC25A46_EAS/EAS_SLC25A46.vcf.gz > SLC25A46_EAS/EAS_SLC25A46.annovar.avinput>\n",
      "NOTICE: Finished reading 216 lines from VCF file\n",
      "NOTICE: A total of 209 locus in VCF file passed QC threshold, representing 193 SNPs (123 transitions and 70 transversions) and 16 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 991827 SNP genotypes (632097 transitions and 359730 transversions) and 82224 indels/substitutions for 5139 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl SLC25A46_EAS/EAS_SLC25A46.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile SLC25A46_EAS/EAS_SLC25A46.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile SLC25A46_EAS/EAS_SLC25A46.annovar.refGene -exonsort -nofirstcodondel SLC25A46_EAS/EAS_SLC25A46.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to SLC25A46_EAS/EAS_SLC25A46.annovar.refGene.variant_function, SLC25A46_EAS/EAS_SLC25A46.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 209 unique variants in 209 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 3 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile SLC25A46_EAS/EAS_SLC25A46.annovar SLC25A46_EAS/EAS_SLC25A46.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to SLC25A46_EAS/EAS_SLC25A46.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to SLC25A46_EAS/EAS_SLC25A46.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 209 unique variants in 209 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to SLC25A46_EAS/EAS_SLC25A46.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from SLC25A46_EAS/EAS_SLC25A46.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to SLC25A46_EAS/EAS_SLC25A46.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_SLC25A46.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/{ancestry}_SLC25A46.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = gene[gene['Gene.refGene'] == 'SLC25A46']\n",
    "gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[(gene['Func.refGene'] == 'exonic')]\n",
    "coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo5142</th>\n",
       "      <th>Otherinfo5143</th>\n",
       "      <th>Otherinfo5144</th>\n",
       "      <th>Otherinfo5145</th>\n",
       "      <th>Otherinfo5146</th>\n",
       "      <th>Otherinfo5147</th>\n",
       "      <th>Otherinfo5148</th>\n",
       "      <th>Otherinfo5149</th>\n",
       "      <th>Otherinfo5150</th>\n",
       "      <th>Otherinfo5151</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5</td>\n",
       "      <td>110739207</td>\n",
       "      <td>110739207</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SLC25A46</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SLC25A46:NM_001303249:exon1:c.G88C:p.A30P,SLC2...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5</td>\n",
       "      <td>110739387</td>\n",
       "      <td>110739387</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SLC25A46</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SLC25A46:NM_001303249:exon1:c.C268A:p.Q90K,SLC...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>5</td>\n",
       "      <td>110761398</td>\n",
       "      <td>110761398</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SLC25A46</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>SLC25A46:NM_001303250:exon8:c.G600C:p.K200N,SL...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 5162 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "13     5  110739207  110739207   G   C       exonic     SLC25A46   \n",
       "15     5  110739387  110739387   C   A       exonic     SLC25A46   \n",
       "183    5  110761398  110761398   G   C       exonic     SLC25A46   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "13                   .  nonsynonymous SNV   \n",
       "15                   .  nonsynonymous SNV   \n",
       "183                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo5142  \\\n",
       "13   SLC25A46:NM_001303249:exon1:c.G88C:p.A30P,SLC2...  ...           0/0   \n",
       "15   SLC25A46:NM_001303249:exon1:c.C268A:p.Q90K,SLC...  ...           0/0   \n",
       "183  SLC25A46:NM_001303250:exon8:c.G600C:p.K200N,SL...  ...           0/0   \n",
       "\n",
       "     Otherinfo5143 Otherinfo5144 Otherinfo5145  Otherinfo5146  Otherinfo5147  \\\n",
       "13             0/0           0/0           0/0            0/0            0/0   \n",
       "15             0/0           0/0           0/0            0/0            0/0   \n",
       "183            0/0           0/0           0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo5148 Otherinfo5149 Otherinfo5150 Otherinfo5151  \n",
       "13            0/0           0/0           0/0           0/0  \n",
       "15            0/0           0/0           0/0           0/0  \n",
       "183           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[3 rows x 5162 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-synonymous variants\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'EAS'}\n",
      "Total variants:  209\n",
      "Intronic:  172\n",
      "UTR3:  25\n",
      "UTR5:  3\n",
      "Total exonic:  9\n",
      "  Synonymous:  6\n",
      "  Nonsynonymous:  3\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "results = [] \n",
    "\n",
    "utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'nonsynonymous SNV')]\n",
    "print({ancestry})\n",
    "print('Total variants: ', len(gene))\n",
    "print(\"Intronic: \", len(intronic))\n",
    "print('UTR3: ', len(utr3))\n",
    "print('UTR5: ', len(utr5))\n",
    "print(\"Total exonic: \", len(exonic))\n",
    "print('  Synonymous: ', len(coding_synonymous))\n",
    "print(\"  Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "results.append((gene, intronic, utr3, utr5, exonic, coding_synonymous, coding_nonsynonymous))\n",
    "print('\\n')\n",
    "    \n",
    "#output = pd.DataFrame(results, columns=('Total variants','Intronic', 'UTR3','UTR5', 'Total exonic', \"Synonymous\", 'Nonsynonymous'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5</td>\n",
       "      <td>110739207</td>\n",
       "      <td>110739207</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5</td>\n",
       "      <td>110739387</td>\n",
       "      <td>110739387</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>5</td>\n",
       "      <td>110761398</td>\n",
       "      <td>110761398</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Gene.refGene\n",
       "13     5  110739207  110739207     SLC25A46\n",
       "15     5  110739387  110739387     SLC25A46\n",
       "183    5  110761398  110761398     SLC25A46"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep = coding_nonsynonymous[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.all_coding_nonsyn.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3, 4)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_toKeep.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5</td>\n",
       "      <td>110739207</td>\n",
       "      <td>110739207</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5</td>\n",
       "      <td>110739266</td>\n",
       "      <td>110739266</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5</td>\n",
       "      <td>110739387</td>\n",
       "      <td>110739387</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>5</td>\n",
       "      <td>110743749</td>\n",
       "      <td>110743749</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>5</td>\n",
       "      <td>110743781</td>\n",
       "      <td>110743781</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>5</td>\n",
       "      <td>110748258</td>\n",
       "      <td>110748258</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>5</td>\n",
       "      <td>110756711</td>\n",
       "      <td>110756711</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>5</td>\n",
       "      <td>110761239</td>\n",
       "      <td>110761239</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>5</td>\n",
       "      <td>110761398</td>\n",
       "      <td>110761398</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Gene.refGene\n",
       "13     5  110739207  110739207     SLC25A46\n",
       "14     5  110739266  110739266     SLC25A46\n",
       "15     5  110739387  110739387     SLC25A46\n",
       "39     5  110743749  110743749     SLC25A46\n",
       "40     5  110743781  110743781     SLC25A46\n",
       "72     5  110748258  110748258     SLC25A46\n",
       "142    5  110756711  110756711     SLC25A46\n",
       "182    5  110761239  110761239     SLC25A46\n",
       "183    5  110761398  110761398     SLC25A46"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep2 = coding[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.all_coding.variantstoKeep', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9, 4)"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_toKeep2.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_SLC25A46.all_coding_nonsyn.variantstoKeep.txt {WORKSPACE_BUCKET}/UHRF1BP1L_{ancestry}/{ancestry}_UHRF1BP1L.all_coding_nonsyn.variantstoKeep ')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_SLC25A46.all_coding.variantstoKeep {WORKSPACE_BUCKET}/UHRF1BP1L_{ancestry}/{ancestry}_UHRF1BP1L.all_coding.variantstoKeep ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case/Control Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Glossary\n",
    "- CHR Chromosome code\n",
    "- SNP Variant identifier\n",
    "- A1 Allele 1 (usually minor)\n",
    "- A2 Allele 2 (usually major)\n",
    "- MAF Allele 1 frequency in all subjects\n",
    "- F_A/MAF_A Allele 1 frequency in cases\n",
    "- F_U/MAF_U Allele 1 frequency in controls\n",
    "- NCHROBS_A Number of case allele observations\n",
    "- NCHROBS_U Number of control allele observations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ALL VARIANTS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Association"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_EAS/EAS_SLC25A46.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile SLC25A46_EAS/EAS_SLC25A46\n",
      "  --ci 0.95\n",
      "  --keep SLC25A46_EAS/EAS.samplestoKeep\n",
      "  --out SLC25A46_EAS/EAS_SLC25A46.all\n",
      "\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\n",
      "1951 variants loaded from .bim file.\n",
      "5139 people (3269 males, 1870 females) loaded from .fam.\n",
      "5099 phenotype values loaded from .fam.\n",
      "--keep: 5138 people remaining.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 5138 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.997895.\n",
      "1951 variants and 5138 people pass filters and QC.\n",
      "Among remaining phenotypes, 2645 are cases and 2453 are controls.  (40\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to SLC25A46_EAS/EAS_SLC25A46.all.assoc ...\n",
      "52done.\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = f'SLC25A46_{ancestry}'\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--assoc \\\n",
    "--allow-no-sex \\\n",
    "--ci 0.95 \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110738976:T:G</td>\n",
       "      <td>110738976</td>\n",
       "      <td>G</td>\n",
       "      <td>0.027300</td>\n",
       "      <td>0.021250</td>\n",
       "      <td>T</td>\n",
       "      <td>3.909</td>\n",
       "      <td>0.048030</td>\n",
       "      <td>1.2930</td>\n",
       "      <td>0.13020</td>\n",
       "      <td>1.00200</td>\n",
       "      <td>1.6690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110739078:G:C</td>\n",
       "      <td>110739078</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001428</td>\n",
       "      <td>G</td>\n",
       "      <td>7.554</td>\n",
       "      <td>0.005989</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110739776:AG:A</td>\n",
       "      <td>110739776</td>\n",
       "      <td>A</td>\n",
       "      <td>0.027270</td>\n",
       "      <td>0.021240</td>\n",
       "      <td>AG</td>\n",
       "      <td>3.886</td>\n",
       "      <td>0.048690</td>\n",
       "      <td>1.2920</td>\n",
       "      <td>0.13020</td>\n",
       "      <td>1.00100</td>\n",
       "      <td>1.6680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110740651:A:G</td>\n",
       "      <td>110740651</td>\n",
       "      <td>G</td>\n",
       "      <td>0.027270</td>\n",
       "      <td>0.021230</td>\n",
       "      <td>A</td>\n",
       "      <td>3.899</td>\n",
       "      <td>0.048320</td>\n",
       "      <td>1.2920</td>\n",
       "      <td>0.13020</td>\n",
       "      <td>1.00100</td>\n",
       "      <td>1.6680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110740674:G:T</td>\n",
       "      <td>110740674</td>\n",
       "      <td>T</td>\n",
       "      <td>0.027270</td>\n",
       "      <td>0.021230</td>\n",
       "      <td>G</td>\n",
       "      <td>3.899</td>\n",
       "      <td>0.048320</td>\n",
       "      <td>1.2920</td>\n",
       "      <td>0.13020</td>\n",
       "      <td>1.00100</td>\n",
       "      <td>1.6680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110744716:G:T</td>\n",
       "      <td>110744716</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001136</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>5.570</td>\n",
       "      <td>0.018280</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>740</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110747478:G:A</td>\n",
       "      <td>110747478</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002840</td>\n",
       "      <td>0.001022</td>\n",
       "      <td>G</td>\n",
       "      <td>4.274</td>\n",
       "      <td>0.038710</td>\n",
       "      <td>2.7820</td>\n",
       "      <td>0.51680</td>\n",
       "      <td>1.01100</td>\n",
       "      <td>7.6610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110749863:T:C</td>\n",
       "      <td>110749863</td>\n",
       "      <td>C</td>\n",
       "      <td>0.184800</td>\n",
       "      <td>0.202200</td>\n",
       "      <td>T</td>\n",
       "      <td>4.884</td>\n",
       "      <td>0.027110</td>\n",
       "      <td>0.8943</td>\n",
       "      <td>0.05054</td>\n",
       "      <td>0.81000</td>\n",
       "      <td>0.9875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110751049:C:A</td>\n",
       "      <td>110751049</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004591</td>\n",
       "      <td>0.007626</td>\n",
       "      <td>C</td>\n",
       "      <td>3.854</td>\n",
       "      <td>0.049630</td>\n",
       "      <td>0.6002</td>\n",
       "      <td>0.26290</td>\n",
       "      <td>0.35850</td>\n",
       "      <td>1.0050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110752280:C:T</td>\n",
       "      <td>110752280</td>\n",
       "      <td>T</td>\n",
       "      <td>0.002857</td>\n",
       "      <td>0.006240</td>\n",
       "      <td>C</td>\n",
       "      <td>6.446</td>\n",
       "      <td>0.011120</td>\n",
       "      <td>0.4564</td>\n",
       "      <td>0.31690</td>\n",
       "      <td>0.24520</td>\n",
       "      <td>0.8492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1074</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110752547:T:G</td>\n",
       "      <td>110752547</td>\n",
       "      <td>G</td>\n",
       "      <td>0.183000</td>\n",
       "      <td>0.199300</td>\n",
       "      <td>T</td>\n",
       "      <td>4.345</td>\n",
       "      <td>0.037120</td>\n",
       "      <td>0.8998</td>\n",
       "      <td>0.05067</td>\n",
       "      <td>0.81470</td>\n",
       "      <td>0.9937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110754208:A:G</td>\n",
       "      <td>110754208</td>\n",
       "      <td>G</td>\n",
       "      <td>0.182100</td>\n",
       "      <td>0.199100</td>\n",
       "      <td>A</td>\n",
       "      <td>4.694</td>\n",
       "      <td>0.030260</td>\n",
       "      <td>0.8961</td>\n",
       "      <td>0.05063</td>\n",
       "      <td>0.81150</td>\n",
       "      <td>0.9896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110754619:C:A</td>\n",
       "      <td>110754619</td>\n",
       "      <td>A</td>\n",
       "      <td>0.182000</td>\n",
       "      <td>0.199300</td>\n",
       "      <td>C</td>\n",
       "      <td>4.917</td>\n",
       "      <td>0.026600</td>\n",
       "      <td>0.8938</td>\n",
       "      <td>0.05063</td>\n",
       "      <td>0.80940</td>\n",
       "      <td>0.9871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1363</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110756460:A:G</td>\n",
       "      <td>110756460</td>\n",
       "      <td>G</td>\n",
       "      <td>0.184300</td>\n",
       "      <td>0.201200</td>\n",
       "      <td>A</td>\n",
       "      <td>4.659</td>\n",
       "      <td>0.030890</td>\n",
       "      <td>0.8969</td>\n",
       "      <td>0.05041</td>\n",
       "      <td>0.81250</td>\n",
       "      <td>0.9901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1387</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "      <td>110756711</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.001637</td>\n",
       "      <td>C</td>\n",
       "      <td>5.980</td>\n",
       "      <td>0.014470</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>1.06100</td>\n",
       "      <td>0.01451</td>\n",
       "      <td>0.9281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1396</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110756831:A:G</td>\n",
       "      <td>110756831</td>\n",
       "      <td>G</td>\n",
       "      <td>0.184300</td>\n",
       "      <td>0.201200</td>\n",
       "      <td>A</td>\n",
       "      <td>4.659</td>\n",
       "      <td>0.030890</td>\n",
       "      <td>0.8969</td>\n",
       "      <td>0.05041</td>\n",
       "      <td>0.81250</td>\n",
       "      <td>0.9901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1423</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110757276:A:C</td>\n",
       "      <td>110757276</td>\n",
       "      <td>C</td>\n",
       "      <td>0.183600</td>\n",
       "      <td>0.200700</td>\n",
       "      <td>A</td>\n",
       "      <td>4.779</td>\n",
       "      <td>0.028800</td>\n",
       "      <td>0.8955</td>\n",
       "      <td>0.05052</td>\n",
       "      <td>0.81100</td>\n",
       "      <td>0.9887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1491</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110758239:A:G</td>\n",
       "      <td>110758239</td>\n",
       "      <td>G</td>\n",
       "      <td>0.185600</td>\n",
       "      <td>0.201400</td>\n",
       "      <td>A</td>\n",
       "      <td>4.015</td>\n",
       "      <td>0.045080</td>\n",
       "      <td>0.9040</td>\n",
       "      <td>0.05036</td>\n",
       "      <td>0.81910</td>\n",
       "      <td>0.9978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1597</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110759729:C:T</td>\n",
       "      <td>110759729</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004811</td>\n",
       "      <td>0.007937</td>\n",
       "      <td>C</td>\n",
       "      <td>3.881</td>\n",
       "      <td>0.048830</td>\n",
       "      <td>0.6043</td>\n",
       "      <td>0.25830</td>\n",
       "      <td>0.36430</td>\n",
       "      <td>1.0030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1645</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110760413:T:A</td>\n",
       "      <td>110760413</td>\n",
       "      <td>A</td>\n",
       "      <td>0.183200</td>\n",
       "      <td>0.199300</td>\n",
       "      <td>T</td>\n",
       "      <td>4.242</td>\n",
       "      <td>0.039430</td>\n",
       "      <td>0.9013</td>\n",
       "      <td>0.05044</td>\n",
       "      <td>0.81650</td>\n",
       "      <td>0.9950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1693</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "      <td>110761239</td>\n",
       "      <td>A</td>\n",
       "      <td>0.183200</td>\n",
       "      <td>0.200300</td>\n",
       "      <td>G</td>\n",
       "      <td>4.757</td>\n",
       "      <td>0.029190</td>\n",
       "      <td>0.8957</td>\n",
       "      <td>0.05050</td>\n",
       "      <td>0.81130</td>\n",
       "      <td>0.9889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1719</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110761857:G:A</td>\n",
       "      <td>110761857</td>\n",
       "      <td>A</td>\n",
       "      <td>0.012690</td>\n",
       "      <td>0.008047</td>\n",
       "      <td>G</td>\n",
       "      <td>5.082</td>\n",
       "      <td>0.024170</td>\n",
       "      <td>1.5840</td>\n",
       "      <td>0.20580</td>\n",
       "      <td>1.05800</td>\n",
       "      <td>2.3710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1783</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110762700:A:G</td>\n",
       "      <td>110762700</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004775</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>A</td>\n",
       "      <td>6.451</td>\n",
       "      <td>0.011090</td>\n",
       "      <td>2.5920</td>\n",
       "      <td>0.38920</td>\n",
       "      <td>1.20900</td>\n",
       "      <td>5.5590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1799</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110762932:T:C</td>\n",
       "      <td>110762932</td>\n",
       "      <td>C</td>\n",
       "      <td>0.182300</td>\n",
       "      <td>0.200600</td>\n",
       "      <td>T</td>\n",
       "      <td>5.423</td>\n",
       "      <td>0.019870</td>\n",
       "      <td>0.8888</td>\n",
       "      <td>0.05063</td>\n",
       "      <td>0.80480</td>\n",
       "      <td>0.9815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1882</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110764363:C:T</td>\n",
       "      <td>110764363</td>\n",
       "      <td>T</td>\n",
       "      <td>0.354100</td>\n",
       "      <td>0.375900</td>\n",
       "      <td>C</td>\n",
       "      <td>4.593</td>\n",
       "      <td>0.032110</td>\n",
       "      <td>0.9100</td>\n",
       "      <td>0.04401</td>\n",
       "      <td>0.83480</td>\n",
       "      <td>0.9920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1918</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110764756:G:C</td>\n",
       "      <td>110764756</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001512</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>G</td>\n",
       "      <td>4.941</td>\n",
       "      <td>0.026220</td>\n",
       "      <td>7.4290</td>\n",
       "      <td>1.06100</td>\n",
       "      <td>0.92880</td>\n",
       "      <td>59.4200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHR                  SNP         BP A1       F_A       F_U  A2  CHISQ  \\\n",
       "85      5   chr5:110738976:T:G  110738976  G  0.027300  0.021250   T  3.909   \n",
       "108     5   chr5:110739078:G:C  110739078  C  0.000000  0.001428   G  7.554   \n",
       "184     5  chr5:110739776:AG:A  110739776  A  0.027270  0.021240  AG  3.886   \n",
       "252     5   chr5:110740651:A:G  110740651  G  0.027270  0.021230   A  3.899   \n",
       "257     5   chr5:110740674:G:T  110740674  T  0.027270  0.021230   G  3.899   \n",
       "543     5   chr5:110744716:G:T  110744716  T  0.001136  0.000000   G  5.570   \n",
       "740     5   chr5:110747478:G:A  110747478  A  0.002840  0.001022   G  4.274   \n",
       "881     5   chr5:110749863:T:C  110749863  C  0.184800  0.202200   T  4.884   \n",
       "965     5   chr5:110751049:C:A  110751049  A  0.004591  0.007626   C  3.854   \n",
       "1058    5   chr5:110752280:C:T  110752280  T  0.002857  0.006240   C  6.446   \n",
       "1074    5   chr5:110752547:T:G  110752547  G  0.183000  0.199300   T  4.345   \n",
       "1216    5   chr5:110754208:A:G  110754208  G  0.182100  0.199100   A  4.694   \n",
       "1245    5   chr5:110754619:C:A  110754619  A  0.182000  0.199300   C  4.917   \n",
       "1363    5   chr5:110756460:A:G  110756460  G  0.184300  0.201200   A  4.659   \n",
       "1387    5   chr5:110756711:C:T  110756711  T  0.000190  0.001637   C  5.980   \n",
       "1396    5   chr5:110756831:A:G  110756831  G  0.184300  0.201200   A  4.659   \n",
       "1423    5   chr5:110757276:A:C  110757276  C  0.183600  0.200700   A  4.779   \n",
       "1491    5   chr5:110758239:A:G  110758239  G  0.185600  0.201400   A  4.015   \n",
       "1597    5   chr5:110759729:C:T  110759729  T  0.004811  0.007937   C  3.881   \n",
       "1645    5   chr5:110760413:T:A  110760413  A  0.183200  0.199300   T  4.242   \n",
       "1693    5   chr5:110761239:G:A  110761239  A  0.183200  0.200300   G  4.757   \n",
       "1719    5   chr5:110761857:G:A  110761857  A  0.012690  0.008047   G  5.082   \n",
       "1783    5   chr5:110762700:A:G  110762700  G  0.004775  0.001847   A  6.451   \n",
       "1799    5   chr5:110762932:T:C  110762932  C  0.182300  0.200600   T  5.423   \n",
       "1882    5   chr5:110764363:C:T  110764363  T  0.354100  0.375900   C  4.593   \n",
       "1918    5   chr5:110764756:G:C  110764756  C  0.001512  0.000204   G  4.941   \n",
       "\n",
       "             P      OR       SE      L95      U95  \n",
       "85    0.048030  1.2930  0.13020  1.00200   1.6690  \n",
       "108   0.005989  0.0000      inf  0.00000      NaN  \n",
       "184   0.048690  1.2920  0.13020  1.00100   1.6680  \n",
       "252   0.048320  1.2920  0.13020  1.00100   1.6680  \n",
       "257   0.048320  1.2920  0.13020  1.00100   1.6680  \n",
       "543   0.018280     NaN      NaN      NaN      NaN  \n",
       "740   0.038710  2.7820  0.51680  1.01100   7.6610  \n",
       "881   0.027110  0.8943  0.05054  0.81000   0.9875  \n",
       "965   0.049630  0.6002  0.26290  0.35850   1.0050  \n",
       "1058  0.011120  0.4564  0.31690  0.24520   0.8492  \n",
       "1074  0.037120  0.8998  0.05067  0.81470   0.9937  \n",
       "1216  0.030260  0.8961  0.05063  0.81150   0.9896  \n",
       "1245  0.026600  0.8938  0.05063  0.80940   0.9871  \n",
       "1363  0.030890  0.8969  0.05041  0.81250   0.9901  \n",
       "1387  0.014470  0.1160  1.06100  0.01451   0.9281  \n",
       "1396  0.030890  0.8969  0.05041  0.81250   0.9901  \n",
       "1423  0.028800  0.8955  0.05052  0.81100   0.9887  \n",
       "1491  0.045080  0.9040  0.05036  0.81910   0.9978  \n",
       "1597  0.048830  0.6043  0.25830  0.36430   1.0030  \n",
       "1645  0.039430  0.9013  0.05044  0.81650   0.9950  \n",
       "1693  0.029190  0.8957  0.05050  0.81130   0.9889  \n",
       "1719  0.024170  1.5840  0.20580  1.05800   2.3710  \n",
       "1783  0.011090  2.5920  0.38920  1.20900   5.5590  \n",
       "1799  0.019870  0.8888  0.05063  0.80480   0.9815  \n",
       "1882  0.032110  0.9100  0.04401  0.83480   0.9920  \n",
       "1918  0.026220  7.4290  1.06100  0.92880  59.4200  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.all.assoc', delim_whitespace=True)\n",
    "sig_all_nonadj = freq[freq['P']<0.05]\n",
    "sig_all_nonadj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get the IDs to extract\n",
    "sig_all_nonadj_id = sig_all_nonadj[['SNP']]\n",
    "sig_all_nonadj_id\n",
    "sig_all_nonadj_id.to_csv(f'{WORK_DIR}/{ancestry}.sig_all_nonadj_id.txt', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5_EAS_release7.log\r\n",
      "chr5_EAS_release7.pgen\r\n",
      "chr5_EAS_release7.psam\r\n",
      "chr5_EAS_release7.pvar\r\n",
      "EAS_covariate_file.txt\r\n",
      "EAS_release7.eigenvec\r\n",
      "EAS_release7.related\r\n",
      "EAS.samplestoKeep\r\n",
      "EAS.sig_all_nonadj_id.txt\r\n",
      "EAS_SLC25A46.all.assoc\r\n",
      "EAS_SLC25A46.all_coding_nonsyn.variantstoKeep.txt\r\n",
      "EAS_SLC25A46.all_coding.variantstoKeep\r\n",
      "EAS_SLC25A46.all.log\r\n",
      "EAS_SLC25A46.annovar.avinput\r\n",
      "EAS_SLC25A46.annovar.hg38_multianno.txt\r\n",
      "EAS_SLC25A46.annovar.hg38_multianno.vcf\r\n",
      "EAS_SLC25A46.bed\r\n",
      "EAS_SLC25A46.bim\r\n",
      "EAS_SLC25A46.fam\r\n",
      "EAS_SLC25A46.log\r\n",
      "EAS_SLC25A46.vcf.gz\r\n",
      "EAS_SLC25A46.vcf.gz.tbi\r\n",
      "master_key_release7_final.csv\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_EAS/EAS_SLC25A46.all.nonadj.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_EAS/EAS_SLC25A46\r\n",
      "  --extract SLC25A46_EAS/EAS.sig_all_nonadj_id.txt\r\n",
      "  --keep SLC25A46_EAS/EAS.samplestoKeep\r\n",
      "  --out SLC25A46_EAS/EAS_SLC25A46.all.nonadj\r\n",
      "  --recode A\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "1951 variants loaded from .bim file.\r\n",
      "5139 people (3269 males, 1870 females) loaded from .fam.\r\n",
      "5099 phenotype values loaded from .fam.\r\n",
      "--extract: 26 variants remaining.\r\n",
      "--keep: 5138 people remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 5138 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "Total genotyping rate in remaining samples is 0.987649.\r\n",
      "26 variants and 5138 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 2645 are cases and 2453 are controls.  (40\r\n",
      "phenotypes are missing.)\r\n",
      "--recode A to SLC25A46_EAS/EAS_SLC25A46.all.nonadj.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A).\n",
    "# Also extract the significant variants \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract {WORK_DIR}/{ancestry}.sig_all_nonadj_id.txt \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.all.nonadj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.all.nonadj.raw', delim_whitespace=True)\n",
    "recode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in EAS for SLC25A46: 26\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr5:110738976:T:G_G',\n",
       " 'chr5:110739078:G:C_C',\n",
       " 'chr5:110739776:AG:A_A',\n",
       " 'chr5:110740651:A:G_G',\n",
       " 'chr5:110740674:G:T_T',\n",
       " 'chr5:110744716:G:T_T',\n",
       " 'chr5:110747478:G:A_A',\n",
       " 'chr5:110749863:T:C_C',\n",
       " 'chr5:110751049:C:A_A',\n",
       " 'chr5:110752280:C:T_T',\n",
       " 'chr5:110752547:T:G_G',\n",
       " 'chr5:110754208:A:G_G',\n",
       " 'chr5:110754619:C:A_A',\n",
       " 'chr5:110756460:A:G_G',\n",
       " 'chr5:110756711:C:T_T',\n",
       " 'chr5:110756831:A:G_G',\n",
       " 'chr5:110757276:A:C_C',\n",
       " 'chr5:110758239:A:G_G',\n",
       " 'chr5:110759729:C:T_T',\n",
       " 'chr5:110760413:T:A_A',\n",
       " 'chr5:110761239:G:A_A',\n",
       " 'chr5:110761857:G:A_A',\n",
       " 'chr5:110762700:A:G_G',\n",
       " 'chr5:110762932:T:C_C',\n",
       " 'chr5:110764363:C:T_T',\n",
       " 'chr5:110764756:G:C_C']"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for SLC25A46: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110738976:T:G_G</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110738976:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110739078:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002854</td>\n",
       "      <td>chr5:110739078:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110739776:AG:A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110739776:AG:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5:110740651:A:G_G</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110740651:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5:110740674:G:T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110740674:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5:110744716:G:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.002268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5:110744716:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5:110747478:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.005671</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002038</td>\n",
       "      <td>chr5:110747478:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5:110749863:T:C_C</td>\n",
       "      <td>88</td>\n",
       "      <td>788</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>97</td>\n",
       "      <td>784</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.359152</td>\n",
       "      <td>chr5:110749863:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5:110751049:C:A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.008696</td>\n",
       "      <td>0</td>\n",
       "      <td>37</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.015084</td>\n",
       "      <td>chr5:110751049:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr5:110752280:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.005671</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.012230</td>\n",
       "      <td>chr5:110752280:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr5:110752547:T:G_G</td>\n",
       "      <td>86</td>\n",
       "      <td>785</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.329301</td>\n",
       "      <td>95</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.356298</td>\n",
       "      <td>chr5:110752547:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr5:110754208:A:G_G</td>\n",
       "      <td>85</td>\n",
       "      <td>787</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.329679</td>\n",
       "      <td>97</td>\n",
       "      <td>777</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.356298</td>\n",
       "      <td>chr5:110754208:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr5:110754619:C:A_A</td>\n",
       "      <td>85</td>\n",
       "      <td>786</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.329301</td>\n",
       "      <td>98</td>\n",
       "      <td>776</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.356298</td>\n",
       "      <td>chr5:110754619:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr5:110756460:A:G_G</td>\n",
       "      <td>87</td>\n",
       "      <td>794</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.333081</td>\n",
       "      <td>98</td>\n",
       "      <td>786</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.360375</td>\n",
       "      <td>chr5:110756460:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr5:110756711:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003261</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr5:110756831:A:G_G</td>\n",
       "      <td>87</td>\n",
       "      <td>794</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.333081</td>\n",
       "      <td>98</td>\n",
       "      <td>786</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.360375</td>\n",
       "      <td>chr5:110756831:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr5:110757276:A:C_C</td>\n",
       "      <td>87</td>\n",
       "      <td>789</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>97</td>\n",
       "      <td>783</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.358744</td>\n",
       "      <td>chr5:110757276:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr5:110758239:A:G_G</td>\n",
       "      <td>89</td>\n",
       "      <td>795</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.334216</td>\n",
       "      <td>98</td>\n",
       "      <td>787</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.360783</td>\n",
       "      <td>chr5:110758239:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr5:110759729:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.009452</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.015084</td>\n",
       "      <td>chr5:110759729:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr5:110760413:T:A_A</td>\n",
       "      <td>89</td>\n",
       "      <td>791</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332703</td>\n",
       "      <td>97</td>\n",
       "      <td>781</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357929</td>\n",
       "      <td>chr5:110760413:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr5:110761239:G:A_A</td>\n",
       "      <td>87</td>\n",
       "      <td>792</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332325</td>\n",
       "      <td>98</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357521</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr5:110761857:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.024197</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.015084</td>\n",
       "      <td>chr5:110761857:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr5:110762700:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.009452</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003669</td>\n",
       "      <td>chr5:110762700:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr5:110762932:T:C_C</td>\n",
       "      <td>86</td>\n",
       "      <td>782</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.328166</td>\n",
       "      <td>96</td>\n",
       "      <td>784</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.358744</td>\n",
       "      <td>chr5:110762932:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr5:110764363:C:T_T</td>\n",
       "      <td>281</td>\n",
       "      <td>1093</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.519471</td>\n",
       "      <td>309</td>\n",
       "      <td>982</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.526294</td>\n",
       "      <td>chr5:110764363:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr5:110764756:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.003025</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110764756:G:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0    chr5:110738976:T:G_G          1        142         2645   \n",
       "1    chr5:110739078:G:C_C          0          0         2645   \n",
       "2   chr5:110739776:AG:A_A          1        142         2645   \n",
       "3    chr5:110740651:A:G_G          1        142         2645   \n",
       "4    chr5:110740674:G:T_T          1        142         2645   \n",
       "5    chr5:110744716:G:T_T          0          6         2645   \n",
       "6    chr5:110747478:G:A_A          0         15         2645   \n",
       "7    chr5:110749863:T:C_C         88        788         2645   \n",
       "8    chr5:110751049:C:A_A          1         22         2645   \n",
       "9    chr5:110752280:C:T_T          0         15         2645   \n",
       "10   chr5:110752547:T:G_G         86        785         2645   \n",
       "11   chr5:110754208:A:G_G         85        787         2645   \n",
       "12   chr5:110754619:C:A_A         85        786         2645   \n",
       "13   chr5:110756460:A:G_G         87        794         2645   \n",
       "14   chr5:110756711:C:T_T          0          1         2645   \n",
       "15   chr5:110756831:A:G_G         87        794         2645   \n",
       "16   chr5:110757276:A:C_C         87        789         2645   \n",
       "17   chr5:110758239:A:G_G         89        795         2645   \n",
       "18   chr5:110759729:C:T_T          0         25         2645   \n",
       "19   chr5:110760413:T:A_A         89        791         2645   \n",
       "20   chr5:110761239:G:A_A         87        792         2645   \n",
       "21   chr5:110761857:G:A_A          0         64         2645   \n",
       "22   chr5:110762700:A:G_G          0         25         2645   \n",
       "23   chr5:110762932:T:C_C         86        782         2645   \n",
       "24   chr5:110764363:C:T_T        281       1093         2645   \n",
       "25   chr5:110764756:G:C_C          0          8         2645   \n",
       "\n",
       "    Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                0.054064             1           102            2453   \n",
       "1                0.000000             0             7            2453   \n",
       "2                0.054064             1           102            2453   \n",
       "3                0.054064             1           102            2453   \n",
       "4                0.054064             1           102            2453   \n",
       "5                0.002268             0             0            2453   \n",
       "6                0.005671             0             5            2453   \n",
       "7                0.331191            97           784            2453   \n",
       "8                0.008696             0            37            2453   \n",
       "9                0.005671             0            30            2453   \n",
       "10               0.329301            95           779            2453   \n",
       "11               0.329679            97           777            2453   \n",
       "12               0.329301            98           776            2453   \n",
       "13               0.333081            98           786            2453   \n",
       "14               0.000378             0             8            2453   \n",
       "15               0.333081            98           786            2453   \n",
       "16               0.331191            97           783            2453   \n",
       "17               0.334216            98           787            2453   \n",
       "18               0.009452             1            36            2453   \n",
       "19               0.332703            97           781            2453   \n",
       "20               0.332325            98           779            2453   \n",
       "21               0.024197             1            36            2453   \n",
       "22               0.009452             0             9            2453   \n",
       "23               0.328166            96           784            2453   \n",
       "24               0.519471           309           982            2453   \n",
       "25               0.003025             0             1            2453   \n",
       "\n",
       "    Carrier freq in Controls                  SNP  \n",
       "0                   0.041989   chr5:110738976:T:G  \n",
       "1                   0.002854   chr5:110739078:G:C  \n",
       "2                   0.041989  chr5:110739776:AG:A  \n",
       "3                   0.041989   chr5:110740651:A:G  \n",
       "4                   0.041989   chr5:110740674:G:T  \n",
       "5                   0.000000   chr5:110744716:G:T  \n",
       "6                   0.002038   chr5:110747478:G:A  \n",
       "7                   0.359152   chr5:110749863:T:C  \n",
       "8                   0.015084   chr5:110751049:C:A  \n",
       "9                   0.012230   chr5:110752280:C:T  \n",
       "10                  0.356298   chr5:110752547:T:G  \n",
       "11                  0.356298   chr5:110754208:A:G  \n",
       "12                  0.356298   chr5:110754619:C:A  \n",
       "13                  0.360375   chr5:110756460:A:G  \n",
       "14                  0.003261   chr5:110756711:C:T  \n",
       "15                  0.360375   chr5:110756831:A:G  \n",
       "16                  0.358744   chr5:110757276:A:C  \n",
       "17                  0.360783   chr5:110758239:A:G  \n",
       "18                  0.015084   chr5:110759729:C:T  \n",
       "19                  0.357929   chr5:110760413:T:A  \n",
       "20                  0.357521   chr5:110761239:G:A  \n",
       "21                  0.015084   chr5:110761857:G:A  \n",
       "22                  0.003669   chr5:110762700:A:G  \n",
       "23                  0.358744   chr5:110762932:T:C  \n",
       "24                  0.526294   chr5:110764363:C:T  \n",
       "25                  0.000408   chr5:110764756:G:C  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['SNP'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>L95</th>\n",
       "      <th>OR</th>\n",
       "      <th>U95</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110738976:T:G_G</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110738976:T:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.027300</td>\n",
       "      <td>0.021250</td>\n",
       "      <td>T</td>\n",
       "      <td>1.00200</td>\n",
       "      <td>1.2930</td>\n",
       "      <td>1.6690</td>\n",
       "      <td>0.048030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110739078:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002854</td>\n",
       "      <td>chr5:110739078:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001428</td>\n",
       "      <td>G</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110739776:AG:A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110739776:AG:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.027270</td>\n",
       "      <td>0.021240</td>\n",
       "      <td>AG</td>\n",
       "      <td>1.00100</td>\n",
       "      <td>1.2920</td>\n",
       "      <td>1.6680</td>\n",
       "      <td>0.048690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5:110740651:A:G_G</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110740651:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.027270</td>\n",
       "      <td>0.021230</td>\n",
       "      <td>A</td>\n",
       "      <td>1.00100</td>\n",
       "      <td>1.2920</td>\n",
       "      <td>1.6680</td>\n",
       "      <td>0.048320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5:110740674:G:T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.054064</td>\n",
       "      <td>1</td>\n",
       "      <td>102</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.041989</td>\n",
       "      <td>chr5:110740674:G:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.027270</td>\n",
       "      <td>0.021230</td>\n",
       "      <td>G</td>\n",
       "      <td>1.00100</td>\n",
       "      <td>1.2920</td>\n",
       "      <td>1.6680</td>\n",
       "      <td>0.048320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5:110744716:G:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.002268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5:110744716:G:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001136</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.018280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5:110747478:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.005671</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002038</td>\n",
       "      <td>chr5:110747478:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002840</td>\n",
       "      <td>0.001022</td>\n",
       "      <td>G</td>\n",
       "      <td>1.01100</td>\n",
       "      <td>2.7820</td>\n",
       "      <td>7.6610</td>\n",
       "      <td>0.038710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5:110749863:T:C_C</td>\n",
       "      <td>88</td>\n",
       "      <td>788</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>97</td>\n",
       "      <td>784</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.359152</td>\n",
       "      <td>chr5:110749863:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.184800</td>\n",
       "      <td>0.202200</td>\n",
       "      <td>T</td>\n",
       "      <td>0.81000</td>\n",
       "      <td>0.8943</td>\n",
       "      <td>0.9875</td>\n",
       "      <td>0.027110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5:110751049:C:A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.008696</td>\n",
       "      <td>0</td>\n",
       "      <td>37</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.015084</td>\n",
       "      <td>chr5:110751049:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004591</td>\n",
       "      <td>0.007626</td>\n",
       "      <td>C</td>\n",
       "      <td>0.35850</td>\n",
       "      <td>0.6002</td>\n",
       "      <td>1.0050</td>\n",
       "      <td>0.049630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr5:110752280:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.005671</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.012230</td>\n",
       "      <td>chr5:110752280:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.002857</td>\n",
       "      <td>0.006240</td>\n",
       "      <td>C</td>\n",
       "      <td>0.24520</td>\n",
       "      <td>0.4564</td>\n",
       "      <td>0.8492</td>\n",
       "      <td>0.011120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr5:110752547:T:G_G</td>\n",
       "      <td>86</td>\n",
       "      <td>785</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.329301</td>\n",
       "      <td>95</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.356298</td>\n",
       "      <td>chr5:110752547:T:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.183000</td>\n",
       "      <td>0.199300</td>\n",
       "      <td>T</td>\n",
       "      <td>0.81470</td>\n",
       "      <td>0.8998</td>\n",
       "      <td>0.9937</td>\n",
       "      <td>0.037120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr5:110754208:A:G_G</td>\n",
       "      <td>85</td>\n",
       "      <td>787</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.329679</td>\n",
       "      <td>97</td>\n",
       "      <td>777</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.356298</td>\n",
       "      <td>chr5:110754208:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.182100</td>\n",
       "      <td>0.199100</td>\n",
       "      <td>A</td>\n",
       "      <td>0.81150</td>\n",
       "      <td>0.8961</td>\n",
       "      <td>0.9896</td>\n",
       "      <td>0.030260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr5:110754619:C:A_A</td>\n",
       "      <td>85</td>\n",
       "      <td>786</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.329301</td>\n",
       "      <td>98</td>\n",
       "      <td>776</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.356298</td>\n",
       "      <td>chr5:110754619:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.182000</td>\n",
       "      <td>0.199300</td>\n",
       "      <td>C</td>\n",
       "      <td>0.80940</td>\n",
       "      <td>0.8938</td>\n",
       "      <td>0.9871</td>\n",
       "      <td>0.026600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr5:110756460:A:G_G</td>\n",
       "      <td>87</td>\n",
       "      <td>794</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.333081</td>\n",
       "      <td>98</td>\n",
       "      <td>786</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.360375</td>\n",
       "      <td>chr5:110756460:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.184300</td>\n",
       "      <td>0.201200</td>\n",
       "      <td>A</td>\n",
       "      <td>0.81250</td>\n",
       "      <td>0.8969</td>\n",
       "      <td>0.9901</td>\n",
       "      <td>0.030890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr5:110756711:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003261</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.001637</td>\n",
       "      <td>C</td>\n",
       "      <td>0.01451</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>0.9281</td>\n",
       "      <td>0.014470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr5:110756831:A:G_G</td>\n",
       "      <td>87</td>\n",
       "      <td>794</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.333081</td>\n",
       "      <td>98</td>\n",
       "      <td>786</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.360375</td>\n",
       "      <td>chr5:110756831:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.184300</td>\n",
       "      <td>0.201200</td>\n",
       "      <td>A</td>\n",
       "      <td>0.81250</td>\n",
       "      <td>0.8969</td>\n",
       "      <td>0.9901</td>\n",
       "      <td>0.030890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr5:110757276:A:C_C</td>\n",
       "      <td>87</td>\n",
       "      <td>789</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.331191</td>\n",
       "      <td>97</td>\n",
       "      <td>783</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.358744</td>\n",
       "      <td>chr5:110757276:A:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.183600</td>\n",
       "      <td>0.200700</td>\n",
       "      <td>A</td>\n",
       "      <td>0.81100</td>\n",
       "      <td>0.8955</td>\n",
       "      <td>0.9887</td>\n",
       "      <td>0.028800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr5:110758239:A:G_G</td>\n",
       "      <td>89</td>\n",
       "      <td>795</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.334216</td>\n",
       "      <td>98</td>\n",
       "      <td>787</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.360783</td>\n",
       "      <td>chr5:110758239:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.185600</td>\n",
       "      <td>0.201400</td>\n",
       "      <td>A</td>\n",
       "      <td>0.81910</td>\n",
       "      <td>0.9040</td>\n",
       "      <td>0.9978</td>\n",
       "      <td>0.045080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr5:110759729:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.009452</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.015084</td>\n",
       "      <td>chr5:110759729:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004811</td>\n",
       "      <td>0.007937</td>\n",
       "      <td>C</td>\n",
       "      <td>0.36430</td>\n",
       "      <td>0.6043</td>\n",
       "      <td>1.0030</td>\n",
       "      <td>0.048830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr5:110760413:T:A_A</td>\n",
       "      <td>89</td>\n",
       "      <td>791</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332703</td>\n",
       "      <td>97</td>\n",
       "      <td>781</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357929</td>\n",
       "      <td>chr5:110760413:T:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.183200</td>\n",
       "      <td>0.199300</td>\n",
       "      <td>T</td>\n",
       "      <td>0.81650</td>\n",
       "      <td>0.9013</td>\n",
       "      <td>0.9950</td>\n",
       "      <td>0.039430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr5:110761239:G:A_A</td>\n",
       "      <td>87</td>\n",
       "      <td>792</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332325</td>\n",
       "      <td>98</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357521</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.183200</td>\n",
       "      <td>0.200300</td>\n",
       "      <td>G</td>\n",
       "      <td>0.81130</td>\n",
       "      <td>0.8957</td>\n",
       "      <td>0.9889</td>\n",
       "      <td>0.029190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr5:110761857:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.024197</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.015084</td>\n",
       "      <td>chr5:110761857:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.012690</td>\n",
       "      <td>0.008047</td>\n",
       "      <td>G</td>\n",
       "      <td>1.05800</td>\n",
       "      <td>1.5840</td>\n",
       "      <td>2.3710</td>\n",
       "      <td>0.024170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr5:110762700:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.009452</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003669</td>\n",
       "      <td>chr5:110762700:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004775</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>A</td>\n",
       "      <td>1.20900</td>\n",
       "      <td>2.5920</td>\n",
       "      <td>5.5590</td>\n",
       "      <td>0.011090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr5:110762932:T:C_C</td>\n",
       "      <td>86</td>\n",
       "      <td>782</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.328166</td>\n",
       "      <td>96</td>\n",
       "      <td>784</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.358744</td>\n",
       "      <td>chr5:110762932:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.182300</td>\n",
       "      <td>0.200600</td>\n",
       "      <td>T</td>\n",
       "      <td>0.80480</td>\n",
       "      <td>0.8888</td>\n",
       "      <td>0.9815</td>\n",
       "      <td>0.019870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr5:110764363:C:T_T</td>\n",
       "      <td>281</td>\n",
       "      <td>1093</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.519471</td>\n",
       "      <td>309</td>\n",
       "      <td>982</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.526294</td>\n",
       "      <td>chr5:110764363:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.354100</td>\n",
       "      <td>0.375900</td>\n",
       "      <td>C</td>\n",
       "      <td>0.83480</td>\n",
       "      <td>0.9100</td>\n",
       "      <td>0.9920</td>\n",
       "      <td>0.032110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr5:110764756:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.003025</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110764756:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001512</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>G</td>\n",
       "      <td>0.92880</td>\n",
       "      <td>7.4290</td>\n",
       "      <td>59.4200</td>\n",
       "      <td>0.026220</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0    chr5:110738976:T:G_G          1        142         2645   \n",
       "1    chr5:110739078:G:C_C          0          0         2645   \n",
       "2   chr5:110739776:AG:A_A          1        142         2645   \n",
       "3    chr5:110740651:A:G_G          1        142         2645   \n",
       "4    chr5:110740674:G:T_T          1        142         2645   \n",
       "5    chr5:110744716:G:T_T          0          6         2645   \n",
       "6    chr5:110747478:G:A_A          0         15         2645   \n",
       "7    chr5:110749863:T:C_C         88        788         2645   \n",
       "8    chr5:110751049:C:A_A          1         22         2645   \n",
       "9    chr5:110752280:C:T_T          0         15         2645   \n",
       "10   chr5:110752547:T:G_G         86        785         2645   \n",
       "11   chr5:110754208:A:G_G         85        787         2645   \n",
       "12   chr5:110754619:C:A_A         85        786         2645   \n",
       "13   chr5:110756460:A:G_G         87        794         2645   \n",
       "14   chr5:110756711:C:T_T          0          1         2645   \n",
       "15   chr5:110756831:A:G_G         87        794         2645   \n",
       "16   chr5:110757276:A:C_C         87        789         2645   \n",
       "17   chr5:110758239:A:G_G         89        795         2645   \n",
       "18   chr5:110759729:C:T_T          0         25         2645   \n",
       "19   chr5:110760413:T:A_A         89        791         2645   \n",
       "20   chr5:110761239:G:A_A         87        792         2645   \n",
       "21   chr5:110761857:G:A_A          0         64         2645   \n",
       "22   chr5:110762700:A:G_G          0         25         2645   \n",
       "23   chr5:110762932:T:C_C         86        782         2645   \n",
       "24   chr5:110764363:C:T_T        281       1093         2645   \n",
       "25   chr5:110764756:G:C_C          0          8         2645   \n",
       "\n",
       "    Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                0.054064             1           102            2453   \n",
       "1                0.000000             0             7            2453   \n",
       "2                0.054064             1           102            2453   \n",
       "3                0.054064             1           102            2453   \n",
       "4                0.054064             1           102            2453   \n",
       "5                0.002268             0             0            2453   \n",
       "6                0.005671             0             5            2453   \n",
       "7                0.331191            97           784            2453   \n",
       "8                0.008696             0            37            2453   \n",
       "9                0.005671             0            30            2453   \n",
       "10               0.329301            95           779            2453   \n",
       "11               0.329679            97           777            2453   \n",
       "12               0.329301            98           776            2453   \n",
       "13               0.333081            98           786            2453   \n",
       "14               0.000378             0             8            2453   \n",
       "15               0.333081            98           786            2453   \n",
       "16               0.331191            97           783            2453   \n",
       "17               0.334216            98           787            2453   \n",
       "18               0.009452             1            36            2453   \n",
       "19               0.332703            97           781            2453   \n",
       "20               0.332325            98           779            2453   \n",
       "21               0.024197             1            36            2453   \n",
       "22               0.009452             0             9            2453   \n",
       "23               0.328166            96           784            2453   \n",
       "24               0.519471           309           982            2453   \n",
       "25               0.003025             0             1            2453   \n",
       "\n",
       "    Carrier freq in Controls                  SNP A1       F_A       F_U  A2  \\\n",
       "0                   0.041989   chr5:110738976:T:G  G  0.027300  0.021250   T   \n",
       "1                   0.002854   chr5:110739078:G:C  C  0.000000  0.001428   G   \n",
       "2                   0.041989  chr5:110739776:AG:A  A  0.027270  0.021240  AG   \n",
       "3                   0.041989   chr5:110740651:A:G  G  0.027270  0.021230   A   \n",
       "4                   0.041989   chr5:110740674:G:T  T  0.027270  0.021230   G   \n",
       "5                   0.000000   chr5:110744716:G:T  T  0.001136  0.000000   G   \n",
       "6                   0.002038   chr5:110747478:G:A  A  0.002840  0.001022   G   \n",
       "7                   0.359152   chr5:110749863:T:C  C  0.184800  0.202200   T   \n",
       "8                   0.015084   chr5:110751049:C:A  A  0.004591  0.007626   C   \n",
       "9                   0.012230   chr5:110752280:C:T  T  0.002857  0.006240   C   \n",
       "10                  0.356298   chr5:110752547:T:G  G  0.183000  0.199300   T   \n",
       "11                  0.356298   chr5:110754208:A:G  G  0.182100  0.199100   A   \n",
       "12                  0.356298   chr5:110754619:C:A  A  0.182000  0.199300   C   \n",
       "13                  0.360375   chr5:110756460:A:G  G  0.184300  0.201200   A   \n",
       "14                  0.003261   chr5:110756711:C:T  T  0.000190  0.001637   C   \n",
       "15                  0.360375   chr5:110756831:A:G  G  0.184300  0.201200   A   \n",
       "16                  0.358744   chr5:110757276:A:C  C  0.183600  0.200700   A   \n",
       "17                  0.360783   chr5:110758239:A:G  G  0.185600  0.201400   A   \n",
       "18                  0.015084   chr5:110759729:C:T  T  0.004811  0.007937   C   \n",
       "19                  0.357929   chr5:110760413:T:A  A  0.183200  0.199300   T   \n",
       "20                  0.357521   chr5:110761239:G:A  A  0.183200  0.200300   G   \n",
       "21                  0.015084   chr5:110761857:G:A  A  0.012690  0.008047   G   \n",
       "22                  0.003669   chr5:110762700:A:G  G  0.004775  0.001847   A   \n",
       "23                  0.358744   chr5:110762932:T:C  C  0.182300  0.200600   T   \n",
       "24                  0.526294   chr5:110764363:C:T  T  0.354100  0.375900   C   \n",
       "25                  0.000408   chr5:110764756:G:C  C  0.001512  0.000204   G   \n",
       "\n",
       "        L95      OR      U95         P  \n",
       "0   1.00200  1.2930   1.6690  0.048030  \n",
       "1   0.00000  0.0000      NaN  0.005989  \n",
       "2   1.00100  1.2920   1.6680  0.048690  \n",
       "3   1.00100  1.2920   1.6680  0.048320  \n",
       "4   1.00100  1.2920   1.6680  0.048320  \n",
       "5       NaN     NaN      NaN  0.018280  \n",
       "6   1.01100  2.7820   7.6610  0.038710  \n",
       "7   0.81000  0.8943   0.9875  0.027110  \n",
       "8   0.35850  0.6002   1.0050  0.049630  \n",
       "9   0.24520  0.4564   0.8492  0.011120  \n",
       "10  0.81470  0.8998   0.9937  0.037120  \n",
       "11  0.81150  0.8961   0.9896  0.030260  \n",
       "12  0.80940  0.8938   0.9871  0.026600  \n",
       "13  0.81250  0.8969   0.9901  0.030890  \n",
       "14  0.01451  0.1160   0.9281  0.014470  \n",
       "15  0.81250  0.8969   0.9901  0.030890  \n",
       "16  0.81100  0.8955   0.9887  0.028800  \n",
       "17  0.81910  0.9040   0.9978  0.045080  \n",
       "18  0.36430  0.6043   1.0030  0.048830  \n",
       "19  0.81650  0.9013   0.9950  0.039430  \n",
       "20  0.81130  0.8957   0.9889  0.029190  \n",
       "21  1.05800  1.5840   2.3710  0.024170  \n",
       "22  1.20900  2.5920   5.5590  0.011090  \n",
       "23  0.80480  0.8888   0.9815  0.019870  \n",
       "24  0.83480  0.9100   0.9920  0.032110  \n",
       "25  0.92880  7.4290  59.4200  0.026220  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge with the assoc file\n",
    "sig_merge = sig_all_nonadj[['SNP','A1','F_A','F_U','A2','L95','OR','U95','P']]\n",
    "merged = pd.merge(df_results, sig_merge, on='SNP', how='right')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "## to CSV\n",
    "merged.to_csv(f'{WORK_DIR}/{ancestry}_all-sign.variants-nonadj.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_all-sign.variants-nonadj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry}_all-sign.variants-nonadj.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "#### Logistic regression (glm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Run with covariates\n",
    "WORK_DIR = f'SLC25A46_{ancestry}'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--glm \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8368</th>\n",
       "      <td>5</td>\n",
       "      <td>110752057</td>\n",
       "      <td>chr5:110752057:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001402</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2496</td>\n",
       "      <td>8.958330</td>\n",
       "      <td>0.883660</td>\n",
       "      <td>2.48125</td>\n",
       "      <td>0.013092</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8464</th>\n",
       "      <td>5</td>\n",
       "      <td>110752280</td>\n",
       "      <td>chr5:110752280:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.005689</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2461</td>\n",
       "      <td>0.272875</td>\n",
       "      <td>0.592551</td>\n",
       "      <td>-2.19178</td>\n",
       "      <td>0.028396</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13104</th>\n",
       "      <td>5</td>\n",
       "      <td>110760334</td>\n",
       "      <td>chr5:110760334:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000805</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2484</td>\n",
       "      <td>0.037578</td>\n",
       "      <td>1.496740</td>\n",
       "      <td>-2.19231</td>\n",
       "      <td>0.028357</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       #CHROM        POS                  ID REF ALT PROVISIONAL_REF? A1  \\\n",
       "8368        5  110752057  chr5:110752057:C:T   C   T                Y  T   \n",
       "8464        5  110752280  chr5:110752280:C:T   C   T                Y  T   \n",
       "13104       5  110760334  chr5:110760334:T:C   T   C                Y  C   \n",
       "\n",
       "      OMITTED   A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE   Z_STAT  \\\n",
       "8368        C  0.001402      N  ADD    2496  8.958330    0.883660  2.48125   \n",
       "8464        C  0.005689      N  ADD    2461  0.272875    0.592551 -2.19178   \n",
       "13104       T  0.000805      Y  ADD    2484  0.037578    1.496740 -2.19231   \n",
       "\n",
       "              P ERRCODE  \n",
       "8368   0.013092       .  \n",
       "8464   0.028396       .  \n",
       "13104  0.028357       .  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "significant = assoc_add[assoc_add['P']<0.05]\n",
    "significant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get the IDs to extract\n",
    "sig_all_adj_id = significant[['ID']]\n",
    "sig_all_adj_id\n",
    "sig_all_adj_id.to_csv(f'{WORK_DIR}/{ancestry}.sig_all_adj_id.txt', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A).\n",
    "# Also extract the significant variants \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract {WORK_DIR}/{ancestry}.sig_all_adj_id.txt \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.all.adj\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.all.adj.raw', delim_whitespace=True)\n",
    "recode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in EAS for SLC25A46: 3\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr5:110752057:C:T_T', 'chr5:110752280:C:T_T', 'chr5:110760334:T:C_C']"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for SLC25A46: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110752057:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.002647</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002038</td>\n",
       "      <td>chr5:110752057:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110752280:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.005671</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.012230</td>\n",
       "      <td>chr5:110752280:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110760334:T:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002854</td>\n",
       "      <td>chr5:110760334:T:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0  chr5:110752057:C:T_T          0          7         2645   \n",
       "1  chr5:110752280:C:T_T          0         15         2645   \n",
       "2  chr5:110760334:T:C_C          0          2         2645   \n",
       "\n",
       "   Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0               0.002647             0             5            2453   \n",
       "1               0.005671             0            30            2453   \n",
       "2               0.000756             0             7            2453   \n",
       "\n",
       "   Carrier freq in Controls                  ID  \n",
       "0                  0.002038  chr5:110752057:C:T  \n",
       "1                  0.012230  chr5:110752280:C:T  \n",
       "2                  0.002854  chr5:110760334:T:C  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['ID'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "      <th>A1</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110752057:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.002647</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002038</td>\n",
       "      <td>chr5:110752057:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001402</td>\n",
       "      <td>2496</td>\n",
       "      <td>8.958330</td>\n",
       "      <td>0.883660</td>\n",
       "      <td>2.48125</td>\n",
       "      <td>0.013092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110752280:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.005671</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.012230</td>\n",
       "      <td>chr5:110752280:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.005689</td>\n",
       "      <td>2461</td>\n",
       "      <td>0.272875</td>\n",
       "      <td>0.592551</td>\n",
       "      <td>-2.19178</td>\n",
       "      <td>0.028396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110760334:T:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.002854</td>\n",
       "      <td>chr5:110760334:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000805</td>\n",
       "      <td>2484</td>\n",
       "      <td>0.037578</td>\n",
       "      <td>1.496740</td>\n",
       "      <td>-2.19231</td>\n",
       "      <td>0.028357</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0  chr5:110752057:C:T_T          0          7         2645   \n",
       "1  chr5:110752280:C:T_T          0         15         2645   \n",
       "2  chr5:110760334:T:C_C          0          2         2645   \n",
       "\n",
       "   Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0               0.002647             0             5            2453   \n",
       "1               0.005671             0            30            2453   \n",
       "2               0.000756             0             7            2453   \n",
       "\n",
       "   Carrier freq in Controls                  ID A1   A1_FREQ  OBS_CT  \\\n",
       "0                  0.002038  chr5:110752057:C:T  T  0.001402    2496   \n",
       "1                  0.012230  chr5:110752280:C:T  T  0.005689    2461   \n",
       "2                  0.002854  chr5:110760334:T:C  C  0.000805    2484   \n",
       "\n",
       "         OR  LOG(OR)_SE   Z_STAT         P  \n",
       "0  8.958330    0.883660  2.48125  0.013092  \n",
       "1  0.272875    0.592551 -2.19178  0.028396  \n",
       "2  0.037578    1.496740 -2.19231  0.028357  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge with the glm file\n",
    "sig_merge = significant[['ID','A1','A1_FREQ','OBS_CT','OR','LOG(OR)_SE','Z_STAT','P']]\n",
    "merged = pd.merge(df_results, sig_merge, on='ID', how='right')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "## to CSV\n",
    "merged.to_csv(f'{WORK_DIR}/{ancestry}_all-sign.variants-adj.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_all-sign.variants-adj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry}_all-sign.variants-adj.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CODING"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Association"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_EAS/EAS_SLC25A46.coding.log.\r\n",
      "Options in effect:\r\n",
      "  --allow-no-sex\r\n",
      "  --assoc\r\n",
      "  --bfile SLC25A46_EAS/EAS_SLC25A46\r\n",
      "  --ci 0.95\r\n",
      "  --extract range SLC25A46_EAS/EAS_SLC25A46.all_coding.variantstoKeep\r\n",
      "  --keep SLC25A46_EAS/EAS.samplestoKeep\r\n",
      "  --out SLC25A46_EAS/EAS_SLC25A46.coding\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "1951 variants loaded from .bim file.\r\n",
      "5139 people (3269 males, 1870 females) loaded from .fam.\r\n",
      "5099 phenotype values loaded from .fam.\r\n",
      "--extract range: 1942 variants excluded.\r\n",
      "--extract range: 9 variants remaining.\r\n",
      "--keep: 5138 people remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 5138 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "Total genotyping rate in remaining samples is 0.997232.\r\n",
      "9 variants and 5138 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 2645 are cases and 2453 are controls.  (40\r\n",
      "phenotypes are missing.)\r\n",
      "Writing C/C --assoc report to SLC25A46_EAS/EAS_SLC25A46.coding.assoc ...\r\n",
      "0%\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = f'SLC25A46_{ancestry}'\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range {WORK_DIR}/{ancestry}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--assoc \\\n",
    "--allow-no-sex \\\n",
    "--ci 0.95 \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110739207:G:C</td>\n",
       "      <td>110739207</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002846</td>\n",
       "      <td>0.95750</td>\n",
       "      <td>0.9273</td>\n",
       "      <td>1.4140</td>\n",
       "      <td>0.05799</td>\n",
       "      <td>14.8300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110739266:C:T</td>\n",
       "      <td>110739266</td>\n",
       "      <td>T</td>\n",
       "      <td>0.014240</td>\n",
       "      <td>0.011640</td>\n",
       "      <td>C</td>\n",
       "      <td>1.343000</td>\n",
       "      <td>0.24640</td>\n",
       "      <td>1.2270</td>\n",
       "      <td>0.1769</td>\n",
       "      <td>0.86760</td>\n",
       "      <td>1.7350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110739387:C:A</td>\n",
       "      <td>110739387</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002763</td>\n",
       "      <td>0.95810</td>\n",
       "      <td>0.9284</td>\n",
       "      <td>1.4140</td>\n",
       "      <td>0.05805</td>\n",
       "      <td>14.8500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110743749:T:C</td>\n",
       "      <td>110743749</td>\n",
       "      <td>C</td>\n",
       "      <td>0.014850</td>\n",
       "      <td>0.013950</td>\n",
       "      <td>T</td>\n",
       "      <td>0.144200</td>\n",
       "      <td>0.70410</td>\n",
       "      <td>1.0660</td>\n",
       "      <td>0.1671</td>\n",
       "      <td>0.76790</td>\n",
       "      <td>1.4780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110743781:A:G</td>\n",
       "      <td>110743781</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.856000</td>\n",
       "      <td>0.17310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110748258:G:A</td>\n",
       "      <td>110748258</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000409</td>\n",
       "      <td>G</td>\n",
       "      <td>2.159000</td>\n",
       "      <td>0.14170</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "      <td>110756711</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.001637</td>\n",
       "      <td>C</td>\n",
       "      <td>5.980000</td>\n",
       "      <td>0.01447</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>1.0610</td>\n",
       "      <td>0.01451</td>\n",
       "      <td>0.9281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "      <td>110761239</td>\n",
       "      <td>A</td>\n",
       "      <td>0.183200</td>\n",
       "      <td>0.200300</td>\n",
       "      <td>G</td>\n",
       "      <td>4.757000</td>\n",
       "      <td>0.02919</td>\n",
       "      <td>0.8957</td>\n",
       "      <td>0.0505</td>\n",
       "      <td>0.81130</td>\n",
       "      <td>0.9889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5:110761398:G:C</td>\n",
       "      <td>110761398</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>G</td>\n",
       "      <td>0.414100</td>\n",
       "      <td>0.51990</td>\n",
       "      <td>0.4634</td>\n",
       "      <td>1.2250</td>\n",
       "      <td>0.04201</td>\n",
       "      <td>5.1120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1       F_A       F_U A2     CHISQ  \\\n",
       "0    5  chr5:110739207:G:C  110739207  C  0.000189  0.000204  G  0.002846   \n",
       "1    5  chr5:110739266:C:T  110739266  T  0.014240  0.011640  C  1.343000   \n",
       "2    5  chr5:110739387:C:A  110739387  A  0.000189  0.000204  C  0.002763   \n",
       "3    5  chr5:110743749:T:C  110743749  C  0.014850  0.013950  T  0.144200   \n",
       "4    5  chr5:110743781:A:G  110743781  G  0.000378  0.000000  A  1.856000   \n",
       "5    5  chr5:110748258:G:A  110748258  A  0.000000  0.000409  G  2.159000   \n",
       "6    5  chr5:110756711:C:T  110756711  T  0.000190  0.001637  C  5.980000   \n",
       "7    5  chr5:110761239:G:A  110761239  A  0.183200  0.200300  G  4.757000   \n",
       "8    5  chr5:110761398:G:C  110761398  C  0.000189  0.000408  G  0.414100   \n",
       "\n",
       "         P      OR      SE      L95      U95  \n",
       "0  0.95750  0.9273  1.4140  0.05799  14.8300  \n",
       "1  0.24640  1.2270  0.1769  0.86760   1.7350  \n",
       "2  0.95810  0.9284  1.4140  0.05805  14.8500  \n",
       "3  0.70410  1.0660  0.1671  0.76790   1.4780  \n",
       "4  0.17310     NaN     NaN      NaN      NaN  \n",
       "5  0.14170  0.0000     inf  0.00000      NaN  \n",
       "6  0.01447  0.1160  1.0610  0.01451   0.9281  \n",
       "7  0.02919  0.8957  0.0505  0.81130   0.9889  \n",
       "8  0.51990  0.4634  1.2250  0.04201   5.1120  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.coding.assoc', delim_whitespace=True)\n",
    "#freq[freq['P']<0.05]\n",
    "freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get the IDs to extract\n",
    "#sig_all_nonadj_id = sig_all_nonadj[['SNP']]\n",
    "#sig_all_nonadj_id\n",
    "#sig_all_nonadj_id.to_csv(f'{WORK_DIR}/{ancestry}.sig_all_nonadj_id.txt', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "## check to make sure file was created and saved\n",
    "#! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_EAS/EAS_SLC25A46.coding.nonadj.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_EAS/EAS_SLC25A46\r\n",
      "  --extract range SLC25A46_EAS/EAS_SLC25A46.all_coding.variantstoKeep\r\n",
      "  --keep SLC25A46_EAS/EAS.samplestoKeep\r\n",
      "  --out SLC25A46_EAS/EAS_SLC25A46.coding.nonadj\r\n",
      "  --recode A\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "1951 variants loaded from .bim file.\r\n",
      "5139 people (3269 males, 1870 females) loaded from .fam.\r\n",
      "5099 phenotype values loaded from .fam.\r\n",
      "--extract range: 1942 variants excluded.\r\n",
      "--extract range: 9 variants remaining.\r\n",
      "--keep: 5138 people remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 5138 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "Total genotyping rate in remaining samples is 0.997232.\r\n",
      "9 variants and 5138 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 2645 are cases and 2453 are controls.  (40\r\n",
      "phenotypes are missing.)\r\n",
      "--recode A to SLC25A46_EAS/EAS_SLC25A46.coding.nonadj.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A).\n",
    "# Also extract the significant variants \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range SLC25A46_{ancestry}/{ancestry}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.coding.nonadj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.coding.nonadj.raw', delim_whitespace=True)\n",
    "recode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in EAS for SLC25A46: 9\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr5:110739207:G:C_C',\n",
       " 'chr5:110739266:C:T_T',\n",
       " 'chr5:110739387:C:A_A',\n",
       " 'chr5:110743749:T:C_C',\n",
       " 'chr5:110743781:A:G_G',\n",
       " 'chr5:110748258:G:A_A',\n",
       " 'chr5:110756711:C:T_T',\n",
       " 'chr5:110761239:G:A_A',\n",
       " 'chr5:110761398:G:C_C']"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for SLC25A46: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110739207:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739207:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110739266:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.028355</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.022829</td>\n",
       "      <td>chr5:110739266:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110739387:C:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739387:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5:110743749:T:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.029490</td>\n",
       "      <td>1</td>\n",
       "      <td>66</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.027313</td>\n",
       "      <td>chr5:110743749:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5:110743781:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5:110743781:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5:110748258:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110748258:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5:110756711:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003261</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5:110761239:G:A_A</td>\n",
       "      <td>87</td>\n",
       "      <td>792</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332325</td>\n",
       "      <td>98</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357521</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5:110761398:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110761398:G:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0  chr5:110739207:G:C_C          0          1         2645   \n",
       "1  chr5:110739266:C:T_T          0         75         2645   \n",
       "2  chr5:110739387:C:A_A          0          1         2645   \n",
       "3  chr5:110743749:T:C_C          0         78         2645   \n",
       "4  chr5:110743781:A:G_G          0          2         2645   \n",
       "5  chr5:110748258:G:A_A          0          0         2645   \n",
       "6  chr5:110756711:C:T_T          0          1         2645   \n",
       "7  chr5:110761239:G:A_A         87        792         2645   \n",
       "8  chr5:110761398:G:C_C          0          1         2645   \n",
       "\n",
       "   Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0               0.000378             0             1            2453   \n",
       "1               0.028355             1            55            2453   \n",
       "2               0.000378             0             1            2453   \n",
       "3               0.029490             1            66            2453   \n",
       "4               0.000756             0             0            2453   \n",
       "5               0.000000             0             2            2453   \n",
       "6               0.000378             0             8            2453   \n",
       "7               0.332325            98           779            2453   \n",
       "8               0.000378             0             2            2453   \n",
       "\n",
       "   Carrier freq in Controls                 SNP  \n",
       "0                  0.000408  chr5:110739207:G:C  \n",
       "1                  0.022829  chr5:110739266:C:T  \n",
       "2                  0.000408  chr5:110739387:C:A  \n",
       "3                  0.027313  chr5:110743749:T:C  \n",
       "4                  0.000000  chr5:110743781:A:G  \n",
       "5                  0.000815  chr5:110748258:G:A  \n",
       "6                  0.003261  chr5:110756711:C:T  \n",
       "7                  0.357521  chr5:110761239:G:A  \n",
       "8                  0.000815  chr5:110761398:G:C  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['SNP'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>L95</th>\n",
       "      <th>OR</th>\n",
       "      <th>U95</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110739207:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739207:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>G</td>\n",
       "      <td>0.05799</td>\n",
       "      <td>0.9273</td>\n",
       "      <td>14.8300</td>\n",
       "      <td>0.95750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110739266:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.028355</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.022829</td>\n",
       "      <td>chr5:110739266:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.014240</td>\n",
       "      <td>0.011640</td>\n",
       "      <td>C</td>\n",
       "      <td>0.86760</td>\n",
       "      <td>1.2270</td>\n",
       "      <td>1.7350</td>\n",
       "      <td>0.24640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110739387:C:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739387:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05805</td>\n",
       "      <td>0.9284</td>\n",
       "      <td>14.8500</td>\n",
       "      <td>0.95810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5:110743749:T:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.029490</td>\n",
       "      <td>1</td>\n",
       "      <td>66</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.027313</td>\n",
       "      <td>chr5:110743749:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.014850</td>\n",
       "      <td>0.013950</td>\n",
       "      <td>T</td>\n",
       "      <td>0.76790</td>\n",
       "      <td>1.0660</td>\n",
       "      <td>1.4780</td>\n",
       "      <td>0.70410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5:110743781:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5:110743781:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.17310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5:110748258:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110748258:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000409</td>\n",
       "      <td>G</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.14170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5:110756711:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003261</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.001637</td>\n",
       "      <td>C</td>\n",
       "      <td>0.01451</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>0.9281</td>\n",
       "      <td>0.01447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5:110761239:G:A_A</td>\n",
       "      <td>87</td>\n",
       "      <td>792</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332325</td>\n",
       "      <td>98</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357521</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.183200</td>\n",
       "      <td>0.200300</td>\n",
       "      <td>G</td>\n",
       "      <td>0.81130</td>\n",
       "      <td>0.8957</td>\n",
       "      <td>0.9889</td>\n",
       "      <td>0.02919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5:110761398:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110761398:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>G</td>\n",
       "      <td>0.04201</td>\n",
       "      <td>0.4634</td>\n",
       "      <td>5.1120</td>\n",
       "      <td>0.51990</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0  chr5:110739207:G:C_C          0          1         2645   \n",
       "1  chr5:110739266:C:T_T          0         75         2645   \n",
       "2  chr5:110739387:C:A_A          0          1         2645   \n",
       "3  chr5:110743749:T:C_C          0         78         2645   \n",
       "4  chr5:110743781:A:G_G          0          2         2645   \n",
       "5  chr5:110748258:G:A_A          0          0         2645   \n",
       "6  chr5:110756711:C:T_T          0          1         2645   \n",
       "7  chr5:110761239:G:A_A         87        792         2645   \n",
       "8  chr5:110761398:G:C_C          0          1         2645   \n",
       "\n",
       "   Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0               0.000378             0             1            2453   \n",
       "1               0.028355             1            55            2453   \n",
       "2               0.000378             0             1            2453   \n",
       "3               0.029490             1            66            2453   \n",
       "4               0.000756             0             0            2453   \n",
       "5               0.000000             0             2            2453   \n",
       "6               0.000378             0             8            2453   \n",
       "7               0.332325            98           779            2453   \n",
       "8               0.000378             0             2            2453   \n",
       "\n",
       "   Carrier freq in Controls                 SNP A1       F_A       F_U A2  \\\n",
       "0                  0.000408  chr5:110739207:G:C  C  0.000189  0.000204  G   \n",
       "1                  0.022829  chr5:110739266:C:T  T  0.014240  0.011640  C   \n",
       "2                  0.000408  chr5:110739387:C:A  A  0.000189  0.000204  C   \n",
       "3                  0.027313  chr5:110743749:T:C  C  0.014850  0.013950  T   \n",
       "4                  0.000000  chr5:110743781:A:G  G  0.000378  0.000000  A   \n",
       "5                  0.000815  chr5:110748258:G:A  A  0.000000  0.000409  G   \n",
       "6                  0.003261  chr5:110756711:C:T  T  0.000190  0.001637  C   \n",
       "7                  0.357521  chr5:110761239:G:A  A  0.183200  0.200300  G   \n",
       "8                  0.000815  chr5:110761398:G:C  C  0.000189  0.000408  G   \n",
       "\n",
       "       L95      OR      U95        P  \n",
       "0  0.05799  0.9273  14.8300  0.95750  \n",
       "1  0.86760  1.2270   1.7350  0.24640  \n",
       "2  0.05805  0.9284  14.8500  0.95810  \n",
       "3  0.76790  1.0660   1.4780  0.70410  \n",
       "4      NaN     NaN      NaN  0.17310  \n",
       "5  0.00000  0.0000      NaN  0.14170  \n",
       "6  0.01451  0.1160   0.9281  0.01447  \n",
       "7  0.81130  0.8957   0.9889  0.02919  \n",
       "8  0.04201  0.4634   5.1120  0.51990  "
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge with the assoc file\n",
    "sig_merge = freq[['SNP','A1','F_A','F_U','A2','L95','OR','U95','P']]\n",
    "merged = pd.merge(df_results, sig_merge, on='SNP', how='right')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "## to CSV\n",
    "merged.to_csv(f'{WORK_DIR}/{ancestry}_coding.variants-nonadj.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_coding.variants-nonadj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry}_coding.variants-nonadj.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Logistic regression (glm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#No significant SNPs with/without AGE as covariate\n",
    "WORK_DIR = f'SLC25A46_{ancestry}'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range {WORK_DIR}/{ancestry}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--glm \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>110739207</td>\n",
       "      <td>chr5:110739207:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2501</td>\n",
       "      <td>1.692520</td>\n",
       "      <td>1.473670</td>\n",
       "      <td>0.357082</td>\n",
       "      <td>0.721030</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>110739266</td>\n",
       "      <td>chr5:110739266:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.014842</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2493</td>\n",
       "      <td>0.968072</td>\n",
       "      <td>0.279820</td>\n",
       "      <td>-0.115963</td>\n",
       "      <td>0.907682</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5</td>\n",
       "      <td>110739387</td>\n",
       "      <td>chr5:110739387:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2498</td>\n",
       "      <td>0.432218</td>\n",
       "      <td>1.636960</td>\n",
       "      <td>-0.512429</td>\n",
       "      <td>0.608351</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>5</td>\n",
       "      <td>110743749</td>\n",
       "      <td>chr5:110743749:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.014889</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2485</td>\n",
       "      <td>0.725545</td>\n",
       "      <td>0.275532</td>\n",
       "      <td>-1.164410</td>\n",
       "      <td>0.244258</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>5</td>\n",
       "      <td>110743781</td>\n",
       "      <td>chr5:110743781:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2499</td>\n",
       "      <td>13.420500</td>\n",
       "      <td>1.635000</td>\n",
       "      <td>1.588240</td>\n",
       "      <td>0.112232</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>5</td>\n",
       "      <td>110748258</td>\n",
       "      <td>chr5:110748258:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000401</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2494</td>\n",
       "      <td>0.580422</td>\n",
       "      <td>1.553570</td>\n",
       "      <td>-0.350162</td>\n",
       "      <td>0.726217</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>5</td>\n",
       "      <td>110756711</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000602</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2490</td>\n",
       "      <td>0.334155</td>\n",
       "      <td>1.339390</td>\n",
       "      <td>-0.818393</td>\n",
       "      <td>0.413133</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>5</td>\n",
       "      <td>110761239</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.193127</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2488</td>\n",
       "      <td>1.105760</td>\n",
       "      <td>0.085167</td>\n",
       "      <td>1.180360</td>\n",
       "      <td>0.237856</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>5</td>\n",
       "      <td>110761398</td>\n",
       "      <td>chr5:110761398:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2502</td>\n",
       "      <td>4.512660</td>\n",
       "      <td>1.640980</td>\n",
       "      <td>0.918286</td>\n",
       "      <td>0.358469</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM        POS                  ID REF ALT PROVISIONAL_REF? A1 OMITTED  \\\n",
       "0        5  110739207  chr5:110739207:G:C   G   C                Y  C       G   \n",
       "8        5  110739266  chr5:110739266:C:T   C   T                Y  T       C   \n",
       "16       5  110739387  chr5:110739387:C:A   C   A                Y  A       C   \n",
       "24       5  110743749  chr5:110743749:T:C   T   C                Y  C       T   \n",
       "32       5  110743781  chr5:110743781:A:G   A   G                Y  G       A   \n",
       "40       5  110748258  chr5:110748258:G:A   G   A                Y  A       G   \n",
       "48       5  110756711  chr5:110756711:C:T   C   T                Y  T       C   \n",
       "56       5  110761239  chr5:110761239:G:A   G   A                Y  A       G   \n",
       "64       5  110761398  chr5:110761398:G:C   G   C                Y  C       G   \n",
       "\n",
       "     A1_FREQ FIRTH? TEST  OBS_CT         OR  LOG(OR)_SE    Z_STAT         P  \\\n",
       "0   0.000400      N  ADD    2501   1.692520    1.473670  0.357082  0.721030   \n",
       "8   0.014842      N  ADD    2493   0.968072    0.279820 -0.115963  0.907682   \n",
       "16  0.000200      Y  ADD    2498   0.432218    1.636960 -0.512429  0.608351   \n",
       "24  0.014889      N  ADD    2485   0.725545    0.275532 -1.164410  0.244258   \n",
       "32  0.000400      Y  ADD    2499  13.420500    1.635000  1.588240  0.112232   \n",
       "40  0.000401      Y  ADD    2494   0.580422    1.553570 -0.350162  0.726217   \n",
       "48  0.000602      N  ADD    2490   0.334155    1.339390 -0.818393  0.413133   \n",
       "56  0.193127      N  ADD    2488   1.105760    0.085167  1.180360  0.237856   \n",
       "64  0.000200      Y  ADD    2502   4.512660    1.640980  0.918286  0.358469   \n",
       "\n",
       "   ERRCODE  \n",
       "0        .  \n",
       "8        .  \n",
       "16       .  \n",
       "24       .  \n",
       "32       .  \n",
       "40       .  \n",
       "48       .  \n",
       "56       .  \n",
       "64       .  "
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.coding.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "#assoc_add[assoc_add['P']<0.05]\n",
    "assoc_add"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_EAS/EAS_SLC25A46.coding.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_EAS/EAS_SLC25A46\r\n",
      "  --extract range SLC25A46_EAS/EAS_SLC25A46.all_coding.variantstoKeep\r\n",
      "  --keep SLC25A46_EAS/EAS.samplestoKeep\r\n",
      "  --out SLC25A46_EAS/EAS_SLC25A46.coding\r\n",
      "  --recode A\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "1951 variants loaded from .bim file.\r\n",
      "5139 people (3269 males, 1870 females) loaded from .fam.\r\n",
      "5099 phenotype values loaded from .fam.\r\n",
      "--extract range: 1942 variants excluded.\r\n",
      "--extract range: 9 variants remaining.\r\n",
      "--keep: 5138 people remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 5138 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "Total genotyping rate in remaining samples is 0.997232.\r\n",
      "9 variants and 5138 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 2645 are cases and 2453 are controls.  (40\r\n",
      "phenotypes are missing.)\r\n",
      "--recode A to SLC25A46_EAS/EAS_SLC25A46.coding.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A). \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry}_SLC25A46\\\n",
    "--extract range {WORK_DIR}/{ancestry}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_SLC25A46.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_SLC25A46.coding.raw', delim_whitespace=True)\n",
    "recode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in EAS for SLC25A46: 9\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr5:110739207:G:C_C',\n",
       " 'chr5:110739266:C:T_T',\n",
       " 'chr5:110739387:C:A_A',\n",
       " 'chr5:110743749:T:C_C',\n",
       " 'chr5:110743781:A:G_G',\n",
       " 'chr5:110748258:G:A_A',\n",
       " 'chr5:110756711:C:T_T',\n",
       " 'chr5:110761239:G:A_A',\n",
       " 'chr5:110761398:G:C_C']"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for SLC25A46: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110739207:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739207:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110739266:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.028355</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.022829</td>\n",
       "      <td>chr5:110739266:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110739387:C:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739387:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5:110743749:T:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.029490</td>\n",
       "      <td>1</td>\n",
       "      <td>66</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.027313</td>\n",
       "      <td>chr5:110743749:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5:110743781:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5:110743781:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5:110748258:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110748258:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5:110756711:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003261</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5:110761239:G:A_A</td>\n",
       "      <td>87</td>\n",
       "      <td>792</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332325</td>\n",
       "      <td>98</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357521</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5:110761398:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110761398:G:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0  chr5:110739207:G:C_C          0          1         2645   \n",
       "1  chr5:110739266:C:T_T          0         75         2645   \n",
       "2  chr5:110739387:C:A_A          0          1         2645   \n",
       "3  chr5:110743749:T:C_C          0         78         2645   \n",
       "4  chr5:110743781:A:G_G          0          2         2645   \n",
       "5  chr5:110748258:G:A_A          0          0         2645   \n",
       "6  chr5:110756711:C:T_T          0          1         2645   \n",
       "7  chr5:110761239:G:A_A         87        792         2645   \n",
       "8  chr5:110761398:G:C_C          0          1         2645   \n",
       "\n",
       "   Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0               0.000378             0             1            2453   \n",
       "1               0.028355             1            55            2453   \n",
       "2               0.000378             0             1            2453   \n",
       "3               0.029490             1            66            2453   \n",
       "4               0.000756             0             0            2453   \n",
       "5               0.000000             0             2            2453   \n",
       "6               0.000378             0             8            2453   \n",
       "7               0.332325            98           779            2453   \n",
       "8               0.000378             0             2            2453   \n",
       "\n",
       "   Carrier freq in Controls                  ID  \n",
       "0                  0.000408  chr5:110739207:G:C  \n",
       "1                  0.022829  chr5:110739266:C:T  \n",
       "2                  0.000408  chr5:110739387:C:A  \n",
       "3                  0.027313  chr5:110743749:T:C  \n",
       "4                  0.000000  chr5:110743781:A:G  \n",
       "5                  0.000815  chr5:110748258:G:A  \n",
       "6                  0.003261  chr5:110756711:C:T  \n",
       "7                  0.357521  chr5:110761239:G:A  \n",
       "8                  0.000815  chr5:110761398:G:C  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['ID'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "      <th>A1</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:110739207:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739207:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>2501</td>\n",
       "      <td>1.692520</td>\n",
       "      <td>1.473670</td>\n",
       "      <td>0.357082</td>\n",
       "      <td>0.721030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:110739266:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.028355</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.022829</td>\n",
       "      <td>chr5:110739266:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.014842</td>\n",
       "      <td>2493</td>\n",
       "      <td>0.968072</td>\n",
       "      <td>0.279820</td>\n",
       "      <td>-0.115963</td>\n",
       "      <td>0.907682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:110739387:C:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>chr5:110739387:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>2498</td>\n",
       "      <td>0.432218</td>\n",
       "      <td>1.636960</td>\n",
       "      <td>-0.512429</td>\n",
       "      <td>0.608351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5:110743749:T:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.029490</td>\n",
       "      <td>1</td>\n",
       "      <td>66</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.027313</td>\n",
       "      <td>chr5:110743749:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.014889</td>\n",
       "      <td>2485</td>\n",
       "      <td>0.725545</td>\n",
       "      <td>0.275532</td>\n",
       "      <td>-1.164410</td>\n",
       "      <td>0.244258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5:110743781:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5:110743781:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>2499</td>\n",
       "      <td>13.420500</td>\n",
       "      <td>1.635000</td>\n",
       "      <td>1.588240</td>\n",
       "      <td>0.112232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5:110748258:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110748258:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000401</td>\n",
       "      <td>2494</td>\n",
       "      <td>0.580422</td>\n",
       "      <td>1.553570</td>\n",
       "      <td>-0.350162</td>\n",
       "      <td>0.726217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5:110756711:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.003261</td>\n",
       "      <td>chr5:110756711:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000602</td>\n",
       "      <td>2490</td>\n",
       "      <td>0.334155</td>\n",
       "      <td>1.339390</td>\n",
       "      <td>-0.818393</td>\n",
       "      <td>0.413133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5:110761239:G:A_A</td>\n",
       "      <td>87</td>\n",
       "      <td>792</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.332325</td>\n",
       "      <td>98</td>\n",
       "      <td>779</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.357521</td>\n",
       "      <td>chr5:110761239:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.193127</td>\n",
       "      <td>2488</td>\n",
       "      <td>1.105760</td>\n",
       "      <td>0.085167</td>\n",
       "      <td>1.180360</td>\n",
       "      <td>0.237856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5:110761398:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2645</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2453</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>chr5:110761398:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>2502</td>\n",
       "      <td>4.512660</td>\n",
       "      <td>1.640980</td>\n",
       "      <td>0.918286</td>\n",
       "      <td>0.358469</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0  chr5:110739207:G:C_C          0          1         2645   \n",
       "1  chr5:110739266:C:T_T          0         75         2645   \n",
       "2  chr5:110739387:C:A_A          0          1         2645   \n",
       "3  chr5:110743749:T:C_C          0         78         2645   \n",
       "4  chr5:110743781:A:G_G          0          2         2645   \n",
       "5  chr5:110748258:G:A_A          0          0         2645   \n",
       "6  chr5:110756711:C:T_T          0          1         2645   \n",
       "7  chr5:110761239:G:A_A         87        792         2645   \n",
       "8  chr5:110761398:G:C_C          0          1         2645   \n",
       "\n",
       "   Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0               0.000378             0             1            2453   \n",
       "1               0.028355             1            55            2453   \n",
       "2               0.000378             0             1            2453   \n",
       "3               0.029490             1            66            2453   \n",
       "4               0.000756             0             0            2453   \n",
       "5               0.000000             0             2            2453   \n",
       "6               0.000378             0             8            2453   \n",
       "7               0.332325            98           779            2453   \n",
       "8               0.000378             0             2            2453   \n",
       "\n",
       "   Carrier freq in Controls                  ID A1   A1_FREQ  OBS_CT  \\\n",
       "0                  0.000408  chr5:110739207:G:C  C  0.000400    2501   \n",
       "1                  0.022829  chr5:110739266:C:T  T  0.014842    2493   \n",
       "2                  0.000408  chr5:110739387:C:A  A  0.000200    2498   \n",
       "3                  0.027313  chr5:110743749:T:C  C  0.014889    2485   \n",
       "4                  0.000000  chr5:110743781:A:G  G  0.000400    2499   \n",
       "5                  0.000815  chr5:110748258:G:A  A  0.000401    2494   \n",
       "6                  0.003261  chr5:110756711:C:T  T  0.000602    2490   \n",
       "7                  0.357521  chr5:110761239:G:A  A  0.193127    2488   \n",
       "8                  0.000815  chr5:110761398:G:C  C  0.000200    2502   \n",
       "\n",
       "          OR  LOG(OR)_SE    Z_STAT         P  \n",
       "0   1.692520    1.473670  0.357082  0.721030  \n",
       "1   0.968072    0.279820 -0.115963  0.907682  \n",
       "2   0.432218    1.636960 -0.512429  0.608351  \n",
       "3   0.725545    0.275532 -1.164410  0.244258  \n",
       "4  13.420500    1.635000  1.588240  0.112232  \n",
       "5   0.580422    1.553570 -0.350162  0.726217  \n",
       "6   0.334155    1.339390 -0.818393  0.413133  \n",
       "7   1.105760    0.085167  1.180360  0.237856  \n",
       "8   4.512660    1.640980  0.918286  0.358469  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge with the glm file\n",
    "sig_merge = assoc_add[['ID','A1','A1_FREQ','OBS_CT','OR','LOG(OR)_SE','Z_STAT','P']]\n",
    "merged = pd.merge(df_results, sig_merge, on='ID', how='right')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "## to CSV\n",
    "merged.to_csv(f'{WORK_DIR}/{ancestry}_coding-variants-adj.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_coding-variants-adj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry}_coding-variants-adj.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "268.372px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
