{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SLC25A46 - Single gene analysis in GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract SLC25A46\n",
    "\n",
    "### 3. Annotate SLC25A46 variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "\n",
    "# Bring some visualization functionality \n",
    "import seaborn as sns  \n",
    "\n",
    "# numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"cloud://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name @ `WORKSPACE_NAME`: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project @ `BILLING_PROJECT_ID`: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data @ `WORKSPACE_BUCKET`: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "## GP2 v7.0\n",
    "## Explicitly define release v7.0 path \n",
    "GP2_RELEASE_PATH = 'path/'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "GP2_META_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_RELEASE_PATH_2 = 'cloud://gp2tier2/release6_21122023'\n",
    "GP2_WGS_PATH = f'{GP2_RELEASE_PATH_2}/wgs'\n",
    "\n",
    "print('GP2 v7.0')\n",
    "print(f'Path to GP2 v7.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 Metadata: {GP2_META_PATH}')\n",
    "print(f'Path to GP2 v6.0 WGS Data: {GP2_WGS_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make 'SLC25A46_WGS' for {All Ancestries}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry = 'all_ancestries'\n",
    "WORK_DIR = f'SLC25A46_WGS_{ancestry}'\n",
    "\n",
    "! mkdir {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install plink 2.0\n",
    "cd /home/jupyter/tools/\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"Plink2 is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget https://s3.amazonaws.com/plink2-assets/plink2_linux_x86_64_20240504.zip\n",
    "\n",
    "unzip -o plink2_linux_x86_64_20240504.zip\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\t\t       plink2\t\t\t\t toy.map\r\n",
      "annovar.latest.tar.gz  plink2_linux_x86_64_20240504.zip  toy.ped\r\n",
      "LICENSE\t\t       plink_linux_x86_64_20190304.zip\r\n",
      "plink\t\t       prettify\r\n"
     ]
    }
   ],
   "source": [
    "! ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remove restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_x86_64_20240504.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Check directory where GP2 Tier 2 data is\n",
    "print(\"List available WGS information in GP2 (broken down by ancestry)\")\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_WGS_PATH}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Show list contents of GP2 Tier 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls cloud://gp2tier2/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Copy WGS genotype GP2 Tier 2 data for SLC25A46 Chromosome 5\n",
    "ancestry = 'all_ancestries'  \n",
    "WORK_DIR = f'SLC25A46_WGS_{ancestry}'\n",
    "\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_WGS_PATH}/var_calling/plink/chr5.psam {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_WGS_PATH}/var_calling/plink/chr5.pvar {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_WGS_PATH}/var_calling/plink/chr5.pgen {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Copy clinical data using master key\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_WGS_PATH}/var_calling/WGS_meta_key.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Copy WGS_GP2_PCA.eigenvec file to directory\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/WGS_GP2_PCA.eigenvec {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Visualize WGS_GP2_PCA.eigenvec using '! head' function\n",
    "! head {WORK_DIR}/WGS_GP2_PCA.eigenvec"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5.pgen  chr5.psam  chr5.pvar  WGS_GP2_PCA.eigenvec  WGS_meta_key.csv\r\n"
     ]
    }
   ],
   "source": [
    "## Check files\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load the master key\n",
    "key = pd.read_csv(f'{WORK_DIR}/WGS_meta_key.csv')\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subsetting to keep only a few columns \n",
    "key = key[['GP2sampleID',\"GP2ID\", 'phenotype', \"pheno_for_qc\",'sex_for_qc', 'age', 'age_of_onset', 'label']]\n",
    "# Rename the columns\n",
    "key.rename(columns = {'GP2sampleID':'IID',\"pheno_for_qc\":\"PHENO\",'sex_for_qc':'SEX','age':'AGE','age_of_onset':'AAO'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##visualize only columns labled with \"PHENO\" for all ancestries\n",
    "key.groupby('label')['PHENO'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "##SELECT THE ANCESTRY GROUP YOU WILL BE WORKING WITH\n",
    "ancestry_GROUP = 'EUR'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Subset to keep ancestry of interest (in this case \"EUR\")\n",
    "covar = key[key['label']==f\"{ancestry_GROUP}\"].copy()\n",
    "covar.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load eigenvec file and visualize data from file\n",
    "PCA = pd.read_csv(f'{WORK_DIR}/WGS_GP2_PCA.eigenvec', sep='\\t')\n",
    "PCA.rename(columns = {\"#FID\":\"FID\"}, inplace = True)\n",
    "PCA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 1273 entries, 0 to 2062\n",
      "Data columns (total 8 columns):\n",
      " #   Column     Non-Null Count  Dtype  \n",
      "---  ------     --------------  -----  \n",
      " 0   IID        1273 non-null   object \n",
      " 1   GP2ID      1273 non-null   object \n",
      " 2   phenotype  1273 non-null   object \n",
      " 3   PHENO      1273 non-null   int64  \n",
      " 4   SEX        1273 non-null   int64  \n",
      " 5   AGE        721 non-null    float64\n",
      " 6   AAO        687 non-null    float64\n",
      " 7   label      1273 non-null   object \n",
      "dtypes: float64(2), int64(2), object(4)\n",
      "memory usage: 89.5+ KB\n"
     ]
    }
   ],
   "source": [
    "## Summarize covar Dataframe\n",
    "covar.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1273 entries, 0 to 1272\n",
      "Data columns (total 19 columns):\n",
      " #   Column     Non-Null Count  Dtype  \n",
      "---  ------     --------------  -----  \n",
      " 0   IID        1273 non-null   object \n",
      " 1   GP2ID      1273 non-null   object \n",
      " 2   phenotype  1273 non-null   object \n",
      " 3   PHENO      1273 non-null   int64  \n",
      " 4   SEX        1273 non-null   int64  \n",
      " 5   AGE        721 non-null    float64\n",
      " 6   AAO        687 non-null    float64\n",
      " 7   label      1273 non-null   object \n",
      " 8   FID        1273 non-null   object \n",
      " 9   PC1        1273 non-null   float64\n",
      " 10  PC2        1273 non-null   float64\n",
      " 11  PC3        1273 non-null   float64\n",
      " 12  PC4        1273 non-null   float64\n",
      " 13  PC5        1273 non-null   float64\n",
      " 14  PC6        1273 non-null   float64\n",
      " 15  PC7        1273 non-null   float64\n",
      " 16  PC8        1273 non-null   float64\n",
      " 17  PC9        1273 non-null   float64\n",
      " 18  PC10       1273 non-null   float64\n",
      "dtypes: float64(12), int64(2), object(5)\n",
      "memory usage: 189.1+ KB\n"
     ]
    }
   ],
   "source": [
    "# Merge covar file from above cells with PCA file from before\n",
    "covar_pca= covar.merge(PCA, on='IID', how='inner')\n",
    "covar_pca.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save covar to workspace\n",
    "covar.to_csv(f'{WORK_DIR}/{WORK_DIR}_COVS.txt', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## check to make sure file is correct/saved\n",
    "covar_pca.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##\n",
    "covar_pca_final = covar_pca[['FID','IID', 'SEX', 'AGE', 'PHENO','PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "covar_pca_final\n",
    "covar_pca_final['PHENO'] = covar_pca_final['PHENO'].fillna(-9)\n",
    "covar_pca_final['AGE'] = covar_pca_final['AGE'].fillna(-9)\n",
    "covar_pca_final['SEX'] = covar_pca_final['SEX'].fillna(-9)\n",
    "\n",
    "covar_pca_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make your covariate file\n",
    "covar_pca_final.to_csv(f'{WORK_DIR}/{ancestry_GROUP}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate SLC25A46! Extract using plink and then annotate using annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_WGS_all_ancestries/chr5.qc.log.\n",
      "Options in effect:\n",
      "  --geno 0.05\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out SLC25A46_WGS_all_ancestries/chr5.qc\n",
      "  --pfile SLC25A46_WGS_all_ancestries/chr5\n",
      "\n",
      "Start time: Tue May 21 21:06:07 2024\n",
      "3676 MiB RAM detected, ~2420 available; reserving 1838 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "2078 samples (760 females, 1318 males; 2078 founders) loaded from\n",
      "SLC25A46_WGS_all_ancestries/chr5.psam.\n",
      "5214395 out of 5425833 variants loaded from\n",
      "SLC25A46_WGS_all_ancestries/chr5.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 1012131415161819202122242526272830313233353637383941424344454748495051535455565759606162646566676870717273747677787980828384858788899091939495969799done.\n",
      "--geno: 143788 variants removed due to missing genotype data.\n",
      "5070607 variants remaining after main filters.\n",
      "Writing SLC25A46_WGS_all_ancestries/chr5.qc.fam ... done.\n",
      "Writing SLC25A46_WGS_all_ancestries/chr5.qc.bim ... done.\n",
      "Writing SLC25A46_WGS_all_ancestries/chr5.qc.bed ... 1011131415161819202122242526272829303233343537383940414344454647495051525455565758606162636566676869717273747677787980828384858788899091939495969799done.\n",
      "End time: Tue May 21 21:08:22 2024\n"
     ]
    }
   ],
   "source": [
    "#Basic QC before extracting\n",
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {WORK_DIR}/chr5  \\\n",
    "--geno 0.05 \\\n",
    "--make-bed \\\n",
    "--max-alleles 2 \\\n",
    "--out {WORK_DIR}/chr5.qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PP2A_AMP_PD.ipynb\t\t   SLC25A46_CAS\r\n",
      "PP2A_GP2_AAC.ipynb\t\t   SLC25A46_EAS\r\n",
      "PP2A_GP2_AFR.ipynb\t\t   SLC25A46_EUR\r\n",
      "PP2A_GP2_AJ.ipynb\t\t   SLC25A46_GP2_Genotyping_Imputed_AAC.ipynb\r\n",
      "PP2A_GP2_AMR.ipynb\t\t   SLC25A46_GP2_Genotyping_Imputed_AFR.ipynb\r\n",
      "PP2A_GP2_CAS.ipynb\t\t   SLC25A46_GP2_Genotyping_Imputed_AJ2.0.ipynb\r\n",
      "PP2A_GP2_EAS.ipynb\t\t   SLC25A46_GP2_Genotyping_Imputed_AJ.ipynb\r\n",
      "PP2A_GP2_EUR.ipynb\t\t   SLC25A46_GP2_Genotyping_Imputed_AMR.ipynb\r\n",
      "PP2A_GP2_MDE.ipynb\t\t   SLC25A46_GP2_Genotyping_Imputed_CAS.ipynb\r\n",
      "PP2A_GP2_SAS.ipynb\t\t   SLC25A46_GP2_Genotyping_Imputed_EAS.ipynb\r\n",
      "Schneider_test_PP2A_GP2_EUR.ipynb  SLC25A46_GP2_Genotyping_Imputed_EUR.ipynb\r\n",
      "SLC25A46_AAC\t\t\t   SLC25A46_GP2_Genotyping_Imputed_MDE.ipynb\r\n",
      "SLC25A46_AFR\t\t\t   SLC25A46_GP2_Genotyping_Imputed_SAS.ipynb\r\n",
      "SLC25A46_AJ\t\t\t   SLC25A46_GP2_WGS.ipynb\r\n",
      "SLC25A46_AMP_PD.ipynb\t\t   SLC25A46_MDE\r\n",
      "SLC25A46_AMR\t\t\t   SLC25A46_WGS_all_ancestries\r\n"
     ]
    }
   ],
   "source": [
    "! ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! awk '{print $1, $2, $5}' SLC25A46_WGS_all_ancestries/EUR_covariate_file.txt > pheno.txt\n",
    "! head pheno.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_WGS_all_ancestries/SLC25A46.log.\n",
      "Options in effect:\n",
      "  --bfile SLC25A46_WGS_all_ancestries/chr5.qc\n",
      "  --chr 5\n",
      "  --from-bp 110738136\n",
      "  --make-bed\n",
      "  --out SLC25A46_WGS_all_ancestries/SLC25A46\n",
      "  --pheno pheno.txt\n",
      "  --to-bp 110765161\n",
      "\n",
      "Start time: Tue May 21 21:08:29 2024\n",
      "3676 MiB RAM detected, ~2429 available; reserving 1838 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "2078 samples (760 females, 1318 males; 2078 founders) loaded from\n",
      "SLC25A46_WGS_all_ancestries/chr5.qc.fam.\n",
      "5070607 variants loaded from SLC25A46_WGS_all_ancestries/chr5.qc.bim.\n",
      "1 binary phenotype loaded (924 cases, 229 controls).\n",
      "735 variants remaining after main filters.\n",
      "Writing SLC25A46_WGS_all_ancestries/SLC25A46.fam ... done.\n",
      "Writing SLC25A46_WGS_all_ancestries/SLC25A46.bim ... done.\n",
      "Writing SLC25A46_WGS_all_ancestries/SLC25A46.bed ... done.\n",
      "End time: Tue May 21 21:08:32 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "\n",
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/chr5.qc \\\n",
    "--chr 5 \\\n",
    "--from-bp 110738136 \\\n",
    "--to-bp 110765161 \\\n",
    "--pheno pheno.txt \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/SLC25A46"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a sample list per ancestry_group to keep only that genetic data\n",
    "\n",
    "sample_list = covar[\"IID\"].tolist()\n",
    "\n",
    "\n",
    "with open('sample_list.txt', 'w') as file:\n",
    "    for iid in sample_list:\n",
    "        file.write('%s\\n' % iid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cp sample_list.txt SLC25A46_WGS_all_ancestries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "! grep -f SLC25A46_WGS_all_ancestries/sample_list.txt  SLC25A46_WGS_all_ancestries/SLC25A46.fam | cut -f1,2 > SLC25A46_WGS_all_ancestries/to_keep.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5.pgen    chr5.qc.log\t     SLC25A46.log\r\n",
      "chr5.psam    EUR_covariate_file.txt  SLC25A46_WGS_all_ancestries_COVS.txt\r\n",
      "chr5.pvar    sample_list.txt\t     to_keep.txt\r\n",
      "chr5.qc.bed  SLC25A46.bed\t     WGS_GP2_PCA.eigenvec\r\n",
      "chr5.qc.bim  SLC25A46.bim\t     WGS_meta_key.csv\r\n",
      "chr5.qc.fam  SLC25A46.fam\r\n"
     ]
    }
   ],
   "source": [
    "! ls SLC25A46_WGS_all_ancestries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1273\r\n"
     ]
    }
   ],
   "source": [
    "! cat SLC25A46_WGS_all_ancestries/to_keep.txt | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_WGS_all_ancestries/SLC25A46\r\n",
      "  --keep SLC25A46_WGS_all_ancestries/to_keep.txt\r\n",
      "  --make-bed\r\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46\r\n",
      "\r\n",
      "Start time: Tue May 21 21:08:38 2024\r\n",
      "3676 MiB RAM detected, ~2465 available; reserving 1838 MiB for main workspace.\r\n",
      "Using 1 compute thread.\r\n",
      "2078 samples (760 females, 1318 males; 2078 founders) loaded from\r\n",
      "SLC25A46_WGS_all_ancestries/SLC25A46.fam.\r\n",
      "735 variants loaded from SLC25A46_WGS_all_ancestries/SLC25A46.bim.\r\n",
      "1 binary phenotype loaded (924 cases, 229 controls).\r\n",
      "--keep: 1273 samples remaining.\r\n",
      "1273 samples (457 females, 816 males; 1273 founders) remaining after main\r\n",
      "filters.\r\n",
      "924 cases and 229 controls remaining after main filters.\r\n",
      "Writing SLC25A46_WGS_all_ancestries/EUR_SLC25A46.fam ... done.\r\n",
      "Writing SLC25A46_WGS_all_ancestries/EUR_SLC25A46.bim ... done.\r\n",
      "Writing SLC25A46_WGS_all_ancestries/EUR_SLC25A46.bed ... 0%\b\bdone.\r\n",
      "End time: Tue May 21 21:08:38 2024\r\n"
     ]
    }
   ],
   "source": [
    "##Extract only the samples per ancestry\n",
    "\n",
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/SLC25A46 \\\n",
    "--keep {WORK_DIR}/to_keep.txt \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.log.\n",
      "Options in effect:\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\n",
      "  --export vcf id-paste=iid\n",
      "  --mac 2\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46\n",
      "\n",
      "Start time: Tue May 21 21:08:39 2024\n",
      "3676 MiB RAM detected, ~2465 available; reserving 1838 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "1273 samples (457 females, 816 males; 1273 founders) loaded from\n",
      "SLC25A46_WGS_all_ancestries/EUR_SLC25A46.fam.\n",
      "735 variants loaded from SLC25A46_WGS_all_ancestries/EUR_SLC25A46.bim.\n",
      "1 binary phenotype loaded (924 cases, 229 controls).\n",
      "Calculating allele frequencies... done.\n",
      "579 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "156 variants remaining after main filters.\n",
      "--export vcf to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.vcf ... 1010111212131414151616171718191920212122232324252526262728282930303132323333343535363737383939404141424243444445464647484849505051515253535455555657575858596060616262636464656666676768696970717172737374757576767778787980808182828383848585868787888989909191929293949495969697989899done.\n",
      "End time: Tue May 21 21:08:39 2024\n"
     ]
    }
   ],
   "source": [
    "## Turn binary files into VCF\n",
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry_GROUP}_SLC25A46.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry_GROUP}_SLC25A46.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 SLC25A46_WGS_all_ancestries/EUR_SLC25A46.vcf.gz > SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.avinput>\n",
      "NOTICE: Finished reading 163 lines from VCF file\n",
      "NOTICE: A total of 156 locus in VCF file passed QC threshold, representing 140 SNPs (95 transitions and 45 transversions) and 16 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 178220 SNP genotypes (120935 transitions and 57285 transversions) and 20368 indels/substitutions for 1273 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.refGene -exonsort -nofirstcodondel SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.refGene.variant_function, SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 156 unique variants in 156 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 3 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 156 unique variants in 156 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry_GROUP}_SLC25A46.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = gene[gene['Gene.refGene'] == 'SLC25A46']\n",
    "gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[(gene['Func.refGene'] == 'exonic')]\n",
    "coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter exonic and non-synonymous variants\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "## EXPORT THIS FOR SUMMARY SHEET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'all_ancestries'}\n",
      "Total variants:  156\n",
      "Intronic:  119\n",
      "UTR3:  25\n",
      "UTR5:  1\n",
      "Total exonic:  11\n",
      "  Synonymous:  6\n",
      "  Nonsynonymous:  5\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "results = [] \n",
    "\n",
    "utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'nonsynonymous SNV')]\n",
    "print({ancestry})\n",
    "print('Total variants: ', len(gene))\n",
    "print(\"Intronic: \", len(intronic))\n",
    "print('UTR3: ', len(utr3))\n",
    "print('UTR5: ', len(utr5))\n",
    "print(\"Total exonic: \", len(exonic))\n",
    "print('  Synonymous: ', len(coding_synonymous))\n",
    "print(\"  Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "results.append((gene, intronic, utr3, utr5, exonic, coding_synonymous, coding_nonsynonymous))\n",
    "print('\\n')\n",
    "    \n",
    "#output = pd.DataFrame(results, columns=('Total variants','Intronic', 'UTR3','UTR5', 'Total exonic', \"Synonymous\", 'Nonsynonymous'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>110739267</td>\n",
       "      <td>110739267</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>110739354</td>\n",
       "      <td>110739354</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>5</td>\n",
       "      <td>110746300</td>\n",
       "      <td>110746300</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5</td>\n",
       "      <td>110756712</td>\n",
       "      <td>110756712</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>5</td>\n",
       "      <td>110761292</td>\n",
       "      <td>110761292</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Gene.refGene\n",
       "4      5  110739267  110739267     SLC25A46\n",
       "5      5  110739354  110739354     SLC25A46\n",
       "47     5  110746300  110746300     SLC25A46\n",
       "101    5  110756712  110756712     SLC25A46\n",
       "129    5  110761292  110761292     SLC25A46"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep = coding_nonsynonymous[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_coding_nonsyn.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5, 4)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_toKeep.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>110739266</td>\n",
       "      <td>110739266</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>110739267</td>\n",
       "      <td>110739267</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>110739354</td>\n",
       "      <td>110739354</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5</td>\n",
       "      <td>110743749</td>\n",
       "      <td>110743749</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>5</td>\n",
       "      <td>110743781</td>\n",
       "      <td>110743781</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>5</td>\n",
       "      <td>110746300</td>\n",
       "      <td>110746300</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>5</td>\n",
       "      <td>110746313</td>\n",
       "      <td>110746313</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>5</td>\n",
       "      <td>110756712</td>\n",
       "      <td>110756712</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>5</td>\n",
       "      <td>110761239</td>\n",
       "      <td>110761239</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>5</td>\n",
       "      <td>110761292</td>\n",
       "      <td>110761292</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>5</td>\n",
       "      <td>110761460</td>\n",
       "      <td>110761460</td>\n",
       "      <td>SLC25A46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Gene.refGene\n",
       "3      5  110739266  110739266     SLC25A46\n",
       "4      5  110739267  110739267     SLC25A46\n",
       "5      5  110739354  110739354     SLC25A46\n",
       "29     5  110743749  110743749     SLC25A46\n",
       "30     5  110743781  110743781     SLC25A46\n",
       "47     5  110746300  110746300     SLC25A46\n",
       "48     5  110746313  110746313     SLC25A46\n",
       "101    5  110756712  110756712     SLC25A46\n",
       "128    5  110761239  110761239     SLC25A46\n",
       "129    5  110761292  110761292     SLC25A46\n",
       "130    5  110761460  110761460     SLC25A46"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep2 = coding[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_coding.variantstoKeep', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(11, 4)"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_toKeep2.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5.pgen\r\n",
      "chr5.psam\r\n",
      "chr5.pvar\r\n",
      "chr5.qc.bed\r\n",
      "chr5.qc.bim\r\n",
      "chr5.qc.fam\r\n",
      "chr5.qc.log\r\n",
      "EUR_covariate_file.txt\r\n",
      "EUR_SLC25A46.all_coding_nonsyn.variantstoKeep.txt\r\n",
      "EUR_SLC25A46.all_coding.variantstoKeep\r\n",
      "EUR_SLC25A46.annovar.avinput\r\n",
      "EUR_SLC25A46.annovar.hg38_multianno.txt\r\n",
      "EUR_SLC25A46.annovar.hg38_multianno.vcf\r\n",
      "EUR_SLC25A46.bed\r\n",
      "EUR_SLC25A46.bim\r\n",
      "EUR_SLC25A46.fam\r\n",
      "EUR_SLC25A46.log\r\n",
      "EUR_SLC25A46.vcf.gz\r\n",
      "EUR_SLC25A46.vcf.gz.tbi\r\n",
      "sample_list.txt\r\n",
      "SLC25A46.bed\r\n",
      "SLC25A46.bim\r\n",
      "SLC25A46.fam\r\n",
      "SLC25A46.log\r\n",
      "SLC25A46_WGS_all_ancestries_COVS.txt\r\n",
      "to_keep.txt\r\n",
      "WGS_GP2_PCA.eigenvec\r\n",
      "WGS_meta_key.csv\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_coding_nonsyn.variantstoKeep.txt {WORKSPACE_BUCKET}/UHRF1BP1L_WGS_all_ancestries/{ancestry_GROUP}_UHRF1BP1L.all_coding_nonsyn.variantstoKeep.txt ')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case/Control Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\n",
      "  --ci 0.95\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all\n",
      "\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\n",
      "735 variants loaded from .bim file.\n",
      "1273 people (816 males, 457 females) loaded from .fam.\n",
      "1153 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 1273 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999594.\n",
      "735 variants and 1273 people pass filters and QC.\n",
      "Among remaining phenotypes, 924 are cases and 229 are controls.  (120\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--assoc \\\n",
    "--allow-no-sex \\\n",
    "--ci 0.95 \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110738462_C_T</td>\n",
       "      <td>110738462</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110739857_A_T</td>\n",
       "      <td>110739857</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>A</td>\n",
       "      <td>8.077</td>\n",
       "      <td>0.004483</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110740250_A_ATT</td>\n",
       "      <td>110740250</td>\n",
       "      <td>ATT</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>A</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110741050_T_G</td>\n",
       "      <td>110741050</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>T</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110741329_C_T</td>\n",
       "      <td>110741329</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001623</td>\n",
       "      <td>0.008734</td>\n",
       "      <td>C</td>\n",
       "      <td>6.131</td>\n",
       "      <td>0.013280</td>\n",
       "      <td>0.1846</td>\n",
       "      <td>0.7656</td>\n",
       "      <td>0.04116</td>\n",
       "      <td>0.8275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110742104_A_G</td>\n",
       "      <td>110742104</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>A</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110742876_C_A</td>\n",
       "      <td>110742876</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>4.032</td>\n",
       "      <td>0.044640</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110743095_C_T</td>\n",
       "      <td>110743095</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002294</td>\n",
       "      <td>C</td>\n",
       "      <td>4.061</td>\n",
       "      <td>0.043870</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110745264_G_C</td>\n",
       "      <td>110745264</td>\n",
       "      <td>C</td>\n",
       "      <td>0.003254</td>\n",
       "      <td>0.010920</td>\n",
       "      <td>G</td>\n",
       "      <td>4.530</td>\n",
       "      <td>0.033300</td>\n",
       "      <td>0.2958</td>\n",
       "      <td>0.6078</td>\n",
       "      <td>0.08986</td>\n",
       "      <td>0.9734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110745378_C_T</td>\n",
       "      <td>110745378</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110745398_A_C</td>\n",
       "      <td>110745398</td>\n",
       "      <td>C</td>\n",
       "      <td>0.008658</td>\n",
       "      <td>0.024020</td>\n",
       "      <td>A</td>\n",
       "      <td>7.483</td>\n",
       "      <td>0.006229</td>\n",
       "      <td>0.3549</td>\n",
       "      <td>0.3952</td>\n",
       "      <td>0.16360</td>\n",
       "      <td>0.7700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110746115_A_G</td>\n",
       "      <td>110746115</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>A</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110746265_A_G</td>\n",
       "      <td>110746265</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>A</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110746845_GA_G</td>\n",
       "      <td>110746845</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>GA</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110749081_T_A</td>\n",
       "      <td>110749081</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>T</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110749521_G_A</td>\n",
       "      <td>110749521</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002203</td>\n",
       "      <td>G</td>\n",
       "      <td>3.967</td>\n",
       "      <td>0.046410</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110750965_A_G</td>\n",
       "      <td>110750965</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>A</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>390</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110752308_C_CA</td>\n",
       "      <td>110752308</td>\n",
       "      <td>CA</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110752372_G_A</td>\n",
       "      <td>110752372</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110752745_G_T</td>\n",
       "      <td>110752745</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110753126_C_A</td>\n",
       "      <td>110753126</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110753933_G_A</td>\n",
       "      <td>110753933</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110755620_A_G</td>\n",
       "      <td>110755620</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>A</td>\n",
       "      <td>4.032</td>\n",
       "      <td>0.044640</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110755641_TG_T</td>\n",
       "      <td>110755641</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>TG</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110755893_T_C</td>\n",
       "      <td>110755893</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>T</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110755980_A_AT;chr5_110755980_AT_A</td>\n",
       "      <td>110755980</td>\n",
       "      <td>ATT</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>AT</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110756802_GT_G</td>\n",
       "      <td>110756802</td>\n",
       "      <td>G</td>\n",
       "      <td>0.008117</td>\n",
       "      <td>0.024020</td>\n",
       "      <td>GT</td>\n",
       "      <td>8.324</td>\n",
       "      <td>0.003912</td>\n",
       "      <td>0.3325</td>\n",
       "      <td>0.4004</td>\n",
       "      <td>0.15170</td>\n",
       "      <td>0.7290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110757099_T_C</td>\n",
       "      <td>110757099</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>T</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110757331_G_C</td>\n",
       "      <td>110757331</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110757885_G_C</td>\n",
       "      <td>110757885</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110758218_A_G</td>\n",
       "      <td>110758218</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>A</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110758322_C_T</td>\n",
       "      <td>110758322</td>\n",
       "      <td>T</td>\n",
       "      <td>0.023270</td>\n",
       "      <td>0.041480</td>\n",
       "      <td>C</td>\n",
       "      <td>4.655</td>\n",
       "      <td>0.030960</td>\n",
       "      <td>0.5504</td>\n",
       "      <td>0.2806</td>\n",
       "      <td>0.31760</td>\n",
       "      <td>0.9539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110759194_G_A</td>\n",
       "      <td>110759194</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>607</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110760368_C_T</td>\n",
       "      <td>110760368</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110761840_A_AGTT</td>\n",
       "      <td>110761840</td>\n",
       "      <td>AGTT</td>\n",
       "      <td>0.000541</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>A</td>\n",
       "      <td>4.135</td>\n",
       "      <td>0.042020</td>\n",
       "      <td>0.1234</td>\n",
       "      <td>1.2260</td>\n",
       "      <td>0.01117</td>\n",
       "      <td>1.3640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110762351_T_C</td>\n",
       "      <td>110762351</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001623</td>\n",
       "      <td>0.008734</td>\n",
       "      <td>T</td>\n",
       "      <td>6.131</td>\n",
       "      <td>0.013280</td>\n",
       "      <td>0.1846</td>\n",
       "      <td>0.7656</td>\n",
       "      <td>0.04116</td>\n",
       "      <td>0.8275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>669</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110762899_G_T</td>\n",
       "      <td>110762899</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>4.032</td>\n",
       "      <td>0.044640</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>701</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110764012_T_C</td>\n",
       "      <td>110764012</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>T</td>\n",
       "      <td>4.037</td>\n",
       "      <td>0.044520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>702</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110764048_G_T</td>\n",
       "      <td>110764048</td>\n",
       "      <td>T</td>\n",
       "      <td>0.081170</td>\n",
       "      <td>0.122300</td>\n",
       "      <td>G</td>\n",
       "      <td>7.622</td>\n",
       "      <td>0.005766</td>\n",
       "      <td>0.6341</td>\n",
       "      <td>0.1661</td>\n",
       "      <td>0.45790</td>\n",
       "      <td>0.8782</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHR                                      SNP         BP    A1       F_A  \\\n",
       "5      5                       chr5_110738462_C_T  110738462     T  0.000000   \n",
       "54     5                       chr5_110739857_A_T  110739857     T  0.000000   \n",
       "64     5                     chr5_110740250_A_ATT  110740250   ATT  0.000000   \n",
       "97     5                       chr5_110741050_T_G  110741050     G  0.000000   \n",
       "105    5                       chr5_110741329_C_T  110741329     T  0.001623   \n",
       "126    5                       chr5_110742104_A_G  110742104     G  0.000000   \n",
       "153    5                       chr5_110742876_C_A  110742876     A  0.000000   \n",
       "160    5                       chr5_110743095_C_T  110743095     T  0.000000   \n",
       "210    5                       chr5_110745264_G_C  110745264     C  0.003254   \n",
       "216    5                       chr5_110745378_C_T  110745378     T  0.000000   \n",
       "218    5                       chr5_110745398_A_C  110745398     C  0.008658   \n",
       "233    5                       chr5_110746115_A_G  110746115     G  0.000000   \n",
       "239    5                       chr5_110746265_A_G  110746265     G  0.000000   \n",
       "254    5                      chr5_110746845_GA_G  110746845     G  0.000000   \n",
       "304    5                       chr5_110749081_T_A  110749081     A  0.000000   \n",
       "311    5                       chr5_110749521_G_A  110749521     A  0.000000   \n",
       "353    5                       chr5_110750965_A_G  110750965     G  0.000000   \n",
       "390    5                      chr5_110752308_C_CA  110752308    CA  0.000000   \n",
       "391    5                       chr5_110752372_G_A  110752372     A  0.000000   \n",
       "396    5                       chr5_110752745_G_T  110752745     T  0.000000   \n",
       "404    5                       chr5_110753126_C_A  110753126     A  0.000000   \n",
       "426    5                       chr5_110753933_G_A  110753933     A  0.000000   \n",
       "472    5                       chr5_110755620_A_G  110755620     G  0.000000   \n",
       "473    5                      chr5_110755641_TG_T  110755641     T  0.000000   \n",
       "481    5                       chr5_110755893_T_C  110755893     C  0.000000   \n",
       "483    5  chr5_110755980_A_AT;chr5_110755980_AT_A  110755980   ATT  0.000000   \n",
       "502    5                      chr5_110756802_GT_G  110756802     G  0.008117   \n",
       "510    5                       chr5_110757099_T_C  110757099     C  0.000000   \n",
       "518    5                       chr5_110757331_G_C  110757331     C  0.000000   \n",
       "533    5                       chr5_110757885_G_C  110757885     C  0.000000   \n",
       "542    5                       chr5_110758218_A_G  110758218     G  0.000000   \n",
       "544    5                       chr5_110758322_C_T  110758322     T  0.023270   \n",
       "577    5                       chr5_110759194_G_A  110759194     A  0.000000   \n",
       "607    5                       chr5_110760368_C_T  110760368     T  0.000000   \n",
       "639    5                    chr5_110761840_A_AGTT  110761840  AGTT  0.000541   \n",
       "653    5                       chr5_110762351_T_C  110762351     C  0.001623   \n",
       "669    5                       chr5_110762899_G_T  110762899     T  0.000000   \n",
       "701    5                       chr5_110764012_T_C  110764012     C  0.000000   \n",
       "702    5                       chr5_110764048_G_T  110764048     T  0.081170   \n",
       "\n",
       "          F_U  A2  CHISQ         P      OR      SE      L95     U95  \n",
       "5    0.002183   C  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "54   0.004367   A  8.077  0.004483  0.0000     inf  0.00000     NaN  \n",
       "64   0.002183   A  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "97   0.002183   T  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "105  0.008734   C  6.131  0.013280  0.1846  0.7656  0.04116  0.8275  \n",
       "126  0.002183   A  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "153  0.002183   C  4.032  0.044640  0.0000     inf  0.00000     NaN  \n",
       "160  0.002294   C  4.061  0.043870  0.0000     inf  0.00000     NaN  \n",
       "210  0.010920   G  4.530  0.033300  0.2958  0.6078  0.08986  0.9734  \n",
       "216  0.002183   C  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "218  0.024020   A  7.483  0.006229  0.3549  0.3952  0.16360  0.7700  \n",
       "233  0.002183   A  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "239  0.002183   A  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "254  0.002183  GA  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "304  0.002183   T  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "311  0.002203   G  3.967  0.046410  0.0000     inf  0.00000     NaN  \n",
       "353  0.002183   A  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "390  0.002183   C  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "391  0.002183   G  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "396  0.002183   G  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "404  0.002183   C  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "426  0.002183   G  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "472  0.002183   A  4.032  0.044640  0.0000     inf  0.00000     NaN  \n",
       "473  0.002183  TG  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "481  0.002183   T  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "483  0.002183  AT  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "502  0.024020  GT  8.324  0.003912  0.3325  0.4004  0.15170  0.7290  \n",
       "510  0.002183   T  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "518  0.002183   G  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "533  0.002183   G  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "542  0.002183   A  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "544  0.041480   C  4.655  0.030960  0.5504  0.2806  0.31760  0.9539  \n",
       "577  0.002183   G  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "607  0.002183   C  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "639  0.004367   A  4.135  0.042020  0.1234  1.2260  0.01117  1.3640  \n",
       "653  0.008734   T  6.131  0.013280  0.1846  0.7656  0.04116  0.8275  \n",
       "669  0.002183   G  4.032  0.044640  0.0000     inf  0.00000     NaN  \n",
       "701  0.002183   T  4.037  0.044520  0.0000     inf  0.00000     NaN  \n",
       "702  0.122300   G  7.622  0.005766  0.6341  0.1661  0.45790  0.8782  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.all.assoc', delim_whitespace=True)\n",
    "sig_all_nonadj = freq[freq['P']<0.05]\n",
    "sig_all_nonadj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get the IDs to extract\n",
    "sig_all_nonadj_id = sig_all_nonadj[['SNP']]\n",
    "sig_all_nonadj_id\n",
    "sig_all_nonadj_id.to_csv(f'{WORK_DIR}/{ancestry_GROUP}.sig_all_nonadj_id.txt', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5.pgen\r\n",
      "chr5.psam\r\n",
      "chr5.pvar\r\n",
      "chr5.qc.bed\r\n",
      "chr5.qc.bim\r\n",
      "chr5.qc.fam\r\n",
      "chr5.qc.log\r\n",
      "EUR_covariate_file.txt\r\n",
      "EUR.sig_all_nonadj_id.txt\r\n",
      "EUR_SLC25A46.all.assoc\r\n",
      "EUR_SLC25A46.all_coding_nonsyn.variantstoKeep.txt\r\n",
      "EUR_SLC25A46.all_coding.variantstoKeep\r\n",
      "EUR_SLC25A46.all.log\r\n",
      "EUR_SLC25A46.annovar.avinput\r\n",
      "EUR_SLC25A46.annovar.hg38_multianno.txt\r\n",
      "EUR_SLC25A46.annovar.hg38_multianno.vcf\r\n",
      "EUR_SLC25A46.bed\r\n",
      "EUR_SLC25A46.bim\r\n",
      "EUR_SLC25A46.fam\r\n",
      "EUR_SLC25A46.log\r\n",
      "EUR_SLC25A46.vcf.gz\r\n",
      "EUR_SLC25A46.vcf.gz.tbi\r\n",
      "sample_list.txt\r\n",
      "SLC25A46.bed\r\n",
      "SLC25A46.bim\r\n",
      "SLC25A46.fam\r\n",
      "SLC25A46.log\r\n",
      "SLC25A46_WGS_all_ancestries_COVS.txt\r\n",
      "to_keep.txt\r\n",
      "WGS_GP2_PCA.eigenvec\r\n",
      "WGS_meta_key.csv\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.nonadj.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\r\n",
      "  --extract SLC25A46_WGS_all_ancestries/EUR.sig_all_nonadj_id.txt\r\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.nonadj\r\n",
      "  --recode A\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "735 variants loaded from .bim file.\r\n",
      "1273 people (816 males, 457 females) loaded from .fam.\r\n",
      "1153 phenotype values loaded from .fam.\r\n",
      "--extract: 39 variants remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 1273 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "Total genotyping rate is 0.998207.\r\n",
      "39 variants and 1273 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 924 are cases and 229 are controls.  (120\r\n",
      "phenotypes are missing.)\r\n",
      "--recode A to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.nonadj.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A).\n",
    "# Also extract the significant variants \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--extract {WORK_DIR}/{ancestry_GROUP}.sig_all_nonadj_id.txt \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all.nonadj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.all.nonadj.raw', delim_whitespace=True)\n",
    "recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_all-sign.variants-nonadj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry_GROUP}_all-sign.variants-nonadj.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry_GROUP}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all_adj.log.\n",
      "Options in effect:\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\n",
      "  --covar SLC25A46_WGS_all_ancestries/EUR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all_adj\n",
      "\n",
      "Start time: Tue May 21 21:09:17 2024\n",
      "3676 MiB RAM detected, ~2414 available; reserving 1838 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "1273 samples (457 females, 816 males; 1273 founders) loaded from\n",
      "SLC25A46_WGS_all_ancestries/EUR_SLC25A46.fam.\n",
      "735 variants loaded from SLC25A46_WGS_all_ancestries/EUR_SLC25A46.bim.\n",
      "1 binary phenotype loaded (924 cases, 229 controls).\n",
      "7 covariates loaded from SLC25A46_WGS_all_ancestries/EUR_covariate_file.txt.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\n",
      "phenotype 'PHENO1'.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Tue May 21 21:09:18 2024\n"
     ]
    }
   ],
   "source": [
    "#Run with covariates\n",
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--glm \\\n",
    "--covar {WORK_DIR}/{ancestry_GROUP}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>5</td>\n",
       "      <td>110738636</td>\n",
       "      <td>chr5_110738636_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.003816</td>\n",
       "      <td>1.90978</td>\n",
       "      <td>-2.91581</td>\n",
       "      <td>0.003548</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>5</td>\n",
       "      <td>110738642</td>\n",
       "      <td>chr5_110738642_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.008948</td>\n",
       "      <td>1.83130</td>\n",
       "      <td>-2.57540</td>\n",
       "      <td>0.010012</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>5</td>\n",
       "      <td>110738661</td>\n",
       "      <td>chr5_110738661_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000322</td>\n",
       "      <td>2.33836</td>\n",
       "      <td>-3.43813</td>\n",
       "      <td>0.000586</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>5</td>\n",
       "      <td>110739062</td>\n",
       "      <td>chr5_110739062_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000135</td>\n",
       "      <td>2.39997</td>\n",
       "      <td>-3.71251</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>5</td>\n",
       "      <td>110739083</td>\n",
       "      <td>chr5_110739083_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.004173</td>\n",
       "      <td>1.95773</td>\n",
       "      <td>-2.79867</td>\n",
       "      <td>0.005131</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5648</th>\n",
       "      <td>5</td>\n",
       "      <td>110764161</td>\n",
       "      <td>chr5_110764161_T_A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.001753</td>\n",
       "      <td>2.06097</td>\n",
       "      <td>-3.07923</td>\n",
       "      <td>0.002075</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5704</th>\n",
       "      <td>5</td>\n",
       "      <td>110764313</td>\n",
       "      <td>chr5_110764313_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.002080</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.020198</td>\n",
       "      <td>1.72390</td>\n",
       "      <td>-2.26358</td>\n",
       "      <td>0.023600</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5760</th>\n",
       "      <td>5</td>\n",
       "      <td>110764649</td>\n",
       "      <td>chr5_110764649_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000394</td>\n",
       "      <td>2.23674</td>\n",
       "      <td>-3.50510</td>\n",
       "      <td>0.000456</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5784</th>\n",
       "      <td>5</td>\n",
       "      <td>110764736</td>\n",
       "      <td>chr5_110764736_T_A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.003593</td>\n",
       "      <td>1.96836</td>\n",
       "      <td>-2.85964</td>\n",
       "      <td>0.004241</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5832</th>\n",
       "      <td>5</td>\n",
       "      <td>110764932</td>\n",
       "      <td>chr5_110764932_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>2.09468</td>\n",
       "      <td>-3.36911</td>\n",
       "      <td>0.000754</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM        POS                  ID REF ALT PROVISIONAL_REF? A1  \\\n",
       "64         5  110738636  chr5_110738636_G_A   G   A                Y  A   \n",
       "72         5  110738642  chr5_110738642_G_A   G   A                Y  A   \n",
       "80         5  110738661  chr5_110738661_G_A   G   A                Y  A   \n",
       "152        5  110739062  chr5_110739062_T_C   T   C                Y  C   \n",
       "168        5  110739083  chr5_110739083_G_A   G   A                Y  A   \n",
       "...      ...        ...                 ...  ..  ..              ... ..   \n",
       "5648       5  110764161  chr5_110764161_T_A   T   A                Y  A   \n",
       "5704       5  110764313  chr5_110764313_T_C   T   C                Y  C   \n",
       "5760       5  110764649  chr5_110764649_T_C   T   C                Y  C   \n",
       "5784       5  110764736  chr5_110764736_T_A   T   A                Y  A   \n",
       "5832       5  110764932  chr5_110764932_C_T   C   T                Y  T   \n",
       "\n",
       "     OMITTED   A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE   Z_STAT  \\\n",
       "64         G  0.000693      Y  ADD     721  0.003816     1.90978 -2.91581   \n",
       "72         G  0.000693      Y  ADD     721  0.008948     1.83130 -2.57540   \n",
       "80         G  0.000693      Y  ADD     721  0.000322     2.33836 -3.43813   \n",
       "152        T  0.000693      Y  ADD     721  0.000135     2.39997 -3.71251   \n",
       "168        G  0.000693      Y  ADD     721  0.004173     1.95773 -2.79867   \n",
       "...      ...       ...    ...  ...     ...       ...         ...      ...   \n",
       "5648       T  0.000693      Y  ADD     721  0.001753     2.06097 -3.07923   \n",
       "5704       T  0.002080      Y  ADD     721  0.020198     1.72390 -2.26358   \n",
       "5760       T  0.000693      Y  ADD     721  0.000394     2.23674 -3.50510   \n",
       "5784       T  0.000693      Y  ADD     721  0.003593     1.96836 -2.85964   \n",
       "5832       C  0.000693      Y  ADD     721  0.000861     2.09468 -3.36911   \n",
       "\n",
       "             P ERRCODE  \n",
       "64    0.003548       .  \n",
       "72    0.010012       .  \n",
       "80    0.000586       .  \n",
       "152   0.000205       .  \n",
       "168   0.005131       .  \n",
       "...        ...     ...  \n",
       "5648  0.002075       .  \n",
       "5704  0.023600       .  \n",
       "5760  0.000456       .  \n",
       "5784  0.004241       .  \n",
       "5832  0.000754       .  \n",
       "\n",
       "[138 rows x 17 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "significant = assoc_add[assoc_add['P']<0.05]\n",
    "significant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get the IDs to extract\n",
    "sig_all_adj_id = significant[['ID']]\n",
    "sig_all_adj_id\n",
    "sig_all_adj_id.to_csv(f'{WORK_DIR}/{ancestry_GROUP}.sig_all_adj_id.txt', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.adj.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\r\n",
      "  --extract SLC25A46_WGS_all_ancestries/EUR.sig_all_adj_id.txt\r\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.adj\r\n",
      "  --recode A\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "735 variants loaded from .bim file.\r\n",
      "1273 people (816 males, 457 females) loaded from .fam.\r\n",
      "1153 phenotype values loaded from .fam.\r\n",
      "--extract: 138 variants remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 1273 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "Total genotyping rate is 0.999397.\r\n",
      "138 variants and 1273 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 924 are cases and 229 are controls.  (120\r\n",
      "phenotypes are missing.)\r\n",
      "--recode A to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all.adj.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A).\n",
    "# Also extract the significant variants \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--extract {WORK_DIR}/{ancestry_GROUP}.sig_all_adj_id.txt \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all.adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.all.adj.raw', delim_whitespace=True)\n",
    "recode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in all_ancestries for SLC25A46: 138\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr5_110738636_G_A_A',\n",
       " 'chr5_110738642_G_A_A',\n",
       " 'chr5_110738661_G_A_A',\n",
       " 'chr5_110739062_T_C_C',\n",
       " 'chr5_110739083_G_A_A',\n",
       " 'chr5_110739114_G_A_A',\n",
       " 'chr5_110739115_C_A_A',\n",
       " 'chr5_110739263_T_C_C',\n",
       " 'chr5_110739296_G_A_A',\n",
       " 'chr5_110739369_G_A_A',\n",
       " 'chr5_110739370_G_A_A',\n",
       " 'chr5_110739397_G_A_A',\n",
       " 'chr5_110739428_G_A_A',\n",
       " 'chr5_110739758_A_C_C',\n",
       " 'chr5_110740023_T_G_G',\n",
       " 'chr5_110740200_G_A_A',\n",
       " 'chr5_110740246_A_T_T',\n",
       " 'chr5_110740263_A_G_G',\n",
       " 'chr5_110740378_G_A_A',\n",
       " 'chr5_110740705_A_C_C',\n",
       " 'chr5_110740722_C_T_T',\n",
       " 'chr5_110740755_T_C_C',\n",
       " 'chr5_110740930_C_T_T',\n",
       " 'chr5_110741279_C_T_T',\n",
       " 'chr5_110741298_A_G_G',\n",
       " 'chr5_110741329_C_T_T',\n",
       " 'chr5_110741773_TG_T_T',\n",
       " 'chr5_110742329_A_G_G',\n",
       " 'chr5_110742508_G_A_A',\n",
       " 'chr5_110742613_A_G_G',\n",
       " 'chr5_110742802_A_G_G',\n",
       " 'chr5_110743179_CAAAATAAAT_C_C',\n",
       " 'chr5_110744486_C_A_A',\n",
       " 'chr5_110744514_G_A_A',\n",
       " 'chr5_110744646_T_C_C',\n",
       " 'chr5_110745009_G_A_A',\n",
       " 'chr5_110745044_G_A_A',\n",
       " 'chr5_110745254_T_G_G',\n",
       " 'chr5_110745394_C_T_T',\n",
       " 'chr5_110745499_C_T_T',\n",
       " 'chr5_110745776_A_G_G',\n",
       " 'chr5_110745984_T_C_C',\n",
       " 'chr5_110746095_A_T_T',\n",
       " 'chr5_110746176_T_G_G',\n",
       " 'chr5_110746300_C_A_A',\n",
       " 'chr5_110746421_T_G_G',\n",
       " 'chr5_110746484_C_T_T',\n",
       " 'chr5_110746670_C_T_T',\n",
       " 'chr5_110747044_A_G_G',\n",
       " 'chr5_110747261_A_G_G',\n",
       " 'chr5_110747290_T_C_C',\n",
       " 'chr5_110747928_T_C_C',\n",
       " 'chr5_110748011_A_G_G',\n",
       " 'chr5_110748269_A_T_T',\n",
       " 'chr5_110748369_A_G_G',\n",
       " 'chr5_110748553_A_G_G',\n",
       " 'chr5_110748740_A_G_G',\n",
       " 'chr5_110748869_C_T_T',\n",
       " 'chr5_110748899_G_C_C',\n",
       " 'chr5_110749073_G_A_A',\n",
       " 'chr5_110749744_G_C_C',\n",
       " 'chr5_110750053_G_A_A',\n",
       " 'chr5_110750201_C_T_T',\n",
       " 'chr5_110750279_G_A_A',\n",
       " 'chr5_110750384_T_TA_TA',\n",
       " 'chr5_110750640_A_G_G',\n",
       " 'chr5_110751119_T_G_G',\n",
       " 'chr5_110751496_C_G_G',\n",
       " 'chr5_110751699_G_A_A',\n",
       " 'chr5_110751847_G_A_A',\n",
       " 'chr5_110751997_A_G_G',\n",
       " 'chr5_110752167_T_C_C',\n",
       " 'chr5_110752176_T_C_C',\n",
       " 'chr5_110752287_C_T_T',\n",
       " 'chr5_110752868_C_T_T',\n",
       " 'chr5_110753434_T_C_C',\n",
       " 'chr5_110753491_A_AT_AT',\n",
       " 'chr5_110753594_C_T_T',\n",
       " 'chr5_110753677_T_A_A',\n",
       " 'chr5_110753727_C_T_T',\n",
       " 'chr5_110754166_C_T_T',\n",
       " 'chr5_110754325_T_C_C',\n",
       " 'chr5_110754505_C_T_T',\n",
       " 'chr5_110754519_CTTCT_C_C',\n",
       " 'chr5_110754573_T_C_C',\n",
       " 'chr5_110754836_G_A_A',\n",
       " 'chr5_110755521_CGTAA_C_C',\n",
       " 'chr5_110755659_G_A_A',\n",
       " 'chr5_110756628_T_G_G',\n",
       " 'chr5_110756685_T_G_G',\n",
       " 'chr5_110756802_GT_G_G',\n",
       " 'chr5_110756890_TCA_T_T',\n",
       " 'chr5_110757064_C_G_G',\n",
       " 'chr5_110757280_G_C_C',\n",
       " 'chr5_110757324_G_A_A',\n",
       " 'chr5_110757371_A_G_G',\n",
       " 'chr5_110758329_T_G_G',\n",
       " 'chr5_110758331_A_G_G',\n",
       " 'chr5_110758393_A_G_G',\n",
       " 'chr5_110758573_C_CA_CA',\n",
       " 'chr5_110758652_G_A_A',\n",
       " 'chr5_110758793_AAAAG_A_A',\n",
       " 'chr5_110758925_C_T_T',\n",
       " 'chr5_110758935_A_T_T',\n",
       " 'chr5_110759027_T_A_A',\n",
       " 'chr5_110759137_T_A_A',\n",
       " 'chr5_110759473_T_C_C',\n",
       " 'chr5_110760426_G_A_A',\n",
       " 'chr5_110760507_GCCCCTGGCTA_G_G',\n",
       " 'chr5_110760838_A_G_G',\n",
       " 'chr5_110760907_AC_A_A',\n",
       " 'chr5_110761079_C_T_T',\n",
       " 'chr5_110761132_TAA_T_T',\n",
       " 'chr5_110761292_A_G_G',\n",
       " 'chr5_110761460_C_CTTA_CTTA',\n",
       " 'chr5_110762066_TTTA_T_T',\n",
       " 'chr5_110762087_A_AAAC_AAAC',\n",
       " 'chr5_110762351_T_C_C',\n",
       " 'chr5_110762523_A_T_T',\n",
       " 'chr5_110762530_C_T_T',\n",
       " 'chr5_110762569_C_T_T',\n",
       " 'chr5_110762615_T_C_C',\n",
       " 'chr5_110762688_T_C_C',\n",
       " 'chr5_110762814_CA_C_C',\n",
       " 'chr5_110763186_T_TG_TG',\n",
       " 'chr5_110763292_G_A_A',\n",
       " 'chr5_110763664_C_T_T',\n",
       " 'chr5_110763819_A_AAATT_AAATT',\n",
       " 'chr5_110763893_C_G_G',\n",
       " 'chr5_110763905_C_T_T',\n",
       " 'chr5_110763970_A_G_G',\n",
       " 'chr5_110764100_G_T_T',\n",
       " 'chr5_110764160_A_AT_AT',\n",
       " 'chr5_110764161_T_A_A',\n",
       " 'chr5_110764313_T_C_C',\n",
       " 'chr5_110764649_T_C_C',\n",
       " 'chr5_110764736_T_A_A',\n",
       " 'chr5_110764932_C_T_T']"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for SLC25A46: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_110738636_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110738636_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5_110738642_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110738642_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5_110738661_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110738661_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5_110739062_T_C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110739062_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5_110739083_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110739083_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>chr5_110764161_T_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>924</td>\n",
       "      <td>0.002165</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764161_T_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>chr5_110764313_T_C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>924</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764313_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>chr5_110764649_T_C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764649_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>chr5_110764736_T_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764736_T_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>chr5_110764932_C_T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764932_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0    chr5_110738636_G_A_A          0          1          924   \n",
       "1    chr5_110738642_G_A_A          0          1          924   \n",
       "2    chr5_110738661_G_A_A          0          1          924   \n",
       "3    chr5_110739062_T_C_C          0          1          924   \n",
       "4    chr5_110739083_G_A_A          0          1          924   \n",
       "..                    ...        ...        ...          ...   \n",
       "133  chr5_110764161_T_A_A          0          2          924   \n",
       "134  chr5_110764313_T_C_C          0          3          924   \n",
       "135  chr5_110764649_T_C_C          0          1          924   \n",
       "136  chr5_110764736_T_A_A          0          1          924   \n",
       "137  chr5_110764932_C_T_T          0          1          924   \n",
       "\n",
       "     Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                 0.001082             0             0             229   \n",
       "1                 0.001082             0             0             229   \n",
       "2                 0.001082             0             0             229   \n",
       "3                 0.001082             0             0             229   \n",
       "4                 0.001082             0             0             229   \n",
       "..                     ...           ...           ...             ...   \n",
       "133               0.002165             0             0             229   \n",
       "134               0.003247             0             0             229   \n",
       "135               0.001082             0             0             229   \n",
       "136               0.001082             0             0             229   \n",
       "137               0.001082             0             0             229   \n",
       "\n",
       "     Carrier freq in Controls                  ID  \n",
       "0                         0.0  chr5_110738636_G_A  \n",
       "1                         0.0  chr5_110738642_G_A  \n",
       "2                         0.0  chr5_110738661_G_A  \n",
       "3                         0.0  chr5_110739062_T_C  \n",
       "4                         0.0  chr5_110739083_G_A  \n",
       "..                        ...                 ...  \n",
       "133                       0.0  chr5_110764161_T_A  \n",
       "134                       0.0  chr5_110764313_T_C  \n",
       "135                       0.0  chr5_110764649_T_C  \n",
       "136                       0.0  chr5_110764736_T_A  \n",
       "137                       0.0  chr5_110764932_C_T  \n",
       "\n",
       "[138 rows x 10 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['ID'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "##EXPORT THIS TO THE ALL-glm [p<0.05] sheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "      <th>A1</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_110738636_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110738636_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.003816</td>\n",
       "      <td>1.90978</td>\n",
       "      <td>-2.91581</td>\n",
       "      <td>0.003548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5_110738642_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110738642_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.008948</td>\n",
       "      <td>1.83130</td>\n",
       "      <td>-2.57540</td>\n",
       "      <td>0.010012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5_110738661_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110738661_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000322</td>\n",
       "      <td>2.33836</td>\n",
       "      <td>-3.43813</td>\n",
       "      <td>0.000586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5_110739062_T_C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110739062_T_C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000135</td>\n",
       "      <td>2.39997</td>\n",
       "      <td>-3.71251</td>\n",
       "      <td>0.000205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5_110739083_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110739083_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.004173</td>\n",
       "      <td>1.95773</td>\n",
       "      <td>-2.79867</td>\n",
       "      <td>0.005131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>chr5_110764161_T_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>924</td>\n",
       "      <td>0.002165</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764161_T_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.001753</td>\n",
       "      <td>2.06097</td>\n",
       "      <td>-3.07923</td>\n",
       "      <td>0.002075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>chr5_110764313_T_C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>924</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764313_T_C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002080</td>\n",
       "      <td>721</td>\n",
       "      <td>0.020198</td>\n",
       "      <td>1.72390</td>\n",
       "      <td>-2.26358</td>\n",
       "      <td>0.023600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>chr5_110764649_T_C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764649_T_C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000394</td>\n",
       "      <td>2.23674</td>\n",
       "      <td>-3.50510</td>\n",
       "      <td>0.000456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>chr5_110764736_T_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764736_T_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.003593</td>\n",
       "      <td>1.96836</td>\n",
       "      <td>-2.85964</td>\n",
       "      <td>0.004241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>chr5_110764932_C_T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.0</td>\n",
       "      <td>chr5_110764932_C_T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>2.09468</td>\n",
       "      <td>-3.36911</td>\n",
       "      <td>0.000754</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0    chr5_110738636_G_A_A          0          1          924   \n",
       "1    chr5_110738642_G_A_A          0          1          924   \n",
       "2    chr5_110738661_G_A_A          0          1          924   \n",
       "3    chr5_110739062_T_C_C          0          1          924   \n",
       "4    chr5_110739083_G_A_A          0          1          924   \n",
       "..                    ...        ...        ...          ...   \n",
       "133  chr5_110764161_T_A_A          0          2          924   \n",
       "134  chr5_110764313_T_C_C          0          3          924   \n",
       "135  chr5_110764649_T_C_C          0          1          924   \n",
       "136  chr5_110764736_T_A_A          0          1          924   \n",
       "137  chr5_110764932_C_T_T          0          1          924   \n",
       "\n",
       "     Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                 0.001082             0             0             229   \n",
       "1                 0.001082             0             0             229   \n",
       "2                 0.001082             0             0             229   \n",
       "3                 0.001082             0             0             229   \n",
       "4                 0.001082             0             0             229   \n",
       "..                     ...           ...           ...             ...   \n",
       "133               0.002165             0             0             229   \n",
       "134               0.003247             0             0             229   \n",
       "135               0.001082             0             0             229   \n",
       "136               0.001082             0             0             229   \n",
       "137               0.001082             0             0             229   \n",
       "\n",
       "     Carrier freq in Controls                  ID A1   A1_FREQ  OBS_CT  \\\n",
       "0                         0.0  chr5_110738636_G_A  A  0.000693     721   \n",
       "1                         0.0  chr5_110738642_G_A  A  0.000693     721   \n",
       "2                         0.0  chr5_110738661_G_A  A  0.000693     721   \n",
       "3                         0.0  chr5_110739062_T_C  C  0.000693     721   \n",
       "4                         0.0  chr5_110739083_G_A  A  0.000693     721   \n",
       "..                        ...                 ... ..       ...     ...   \n",
       "133                       0.0  chr5_110764161_T_A  A  0.000693     721   \n",
       "134                       0.0  chr5_110764313_T_C  C  0.002080     721   \n",
       "135                       0.0  chr5_110764649_T_C  C  0.000693     721   \n",
       "136                       0.0  chr5_110764736_T_A  A  0.000693     721   \n",
       "137                       0.0  chr5_110764932_C_T  T  0.000693     721   \n",
       "\n",
       "           OR  LOG(OR)_SE   Z_STAT         P  \n",
       "0    0.003816     1.90978 -2.91581  0.003548  \n",
       "1    0.008948     1.83130 -2.57540  0.010012  \n",
       "2    0.000322     2.33836 -3.43813  0.000586  \n",
       "3    0.000135     2.39997 -3.71251  0.000205  \n",
       "4    0.004173     1.95773 -2.79867  0.005131  \n",
       "..        ...         ...      ...       ...  \n",
       "133  0.001753     2.06097 -3.07923  0.002075  \n",
       "134  0.020198     1.72390 -2.26358  0.023600  \n",
       "135  0.000394     2.23674 -3.50510  0.000456  \n",
       "136  0.003593     1.96836 -2.85964  0.004241  \n",
       "137  0.000861     2.09468 -3.36911  0.000754  \n",
       "\n",
       "[138 rows x 17 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge with the glm file\n",
    "sig_merge = significant[['ID','A1','A1_FREQ','OBS_CT','OR','LOG(OR)_SE','Z_STAT','P']]\n",
    "merged = pd.merge(df_results, sig_merge, on='ID', how='right')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "## to CSV\n",
    "merged.to_csv(f'{WORK_DIR}/{ancestry_GROUP}_all-sign.variants-adj.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_all-sign.variants-adj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry_GROUP}_all-sign.variants-adj.txt')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "CODING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.log.\r\n",
      "Options in effect:\r\n",
      "  --allow-no-sex\r\n",
      "  --assoc\r\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\r\n",
      "  --ci 0.95\r\n",
      "  --extract range SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all_coding.variantstoKeep\r\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "735 variants loaded from .bim file.\r\n",
      "1273 people (816 males, 457 females) loaded from .fam.\r\n",
      "1153 phenotype values loaded from .fam.\r\n",
      "--extract range: 724 variants excluded.\r\n",
      "--extract range: 11 variants remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 1273 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "11 variants and 1273 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 924 are cases and 229 are controls.  (120\r\n",
      "phenotypes are missing.)\r\n",
      "Writing C/C --assoc report to\r\n",
      "SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.assoc ...\r\n",
      "0%\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--extract range {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--assoc \\\n",
    "--allow-no-sex \\\n",
    "--ci 0.95 \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110739266_C_T</td>\n",
       "      <td>110739266</td>\n",
       "      <td>T</td>\n",
       "      <td>0.015690</td>\n",
       "      <td>0.019650</td>\n",
       "      <td>C</td>\n",
       "      <td>0.3548</td>\n",
       "      <td>0.55140</td>\n",
       "      <td>0.7954</td>\n",
       "      <td>0.3852</td>\n",
       "      <td>0.37380</td>\n",
       "      <td>1.692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110739267_C_A</td>\n",
       "      <td>110739267</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000541</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>1.1420</td>\n",
       "      <td>0.28520</td>\n",
       "      <td>0.2474</td>\n",
       "      <td>1.4150</td>\n",
       "      <td>0.01545</td>\n",
       "      <td>3.963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110739354_G_A</td>\n",
       "      <td>110739354</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003788</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>1.7400</td>\n",
       "      <td>0.18710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110743749_T_C</td>\n",
       "      <td>110743749</td>\n",
       "      <td>C</td>\n",
       "      <td>0.087120</td>\n",
       "      <td>0.109200</td>\n",
       "      <td>T</td>\n",
       "      <td>2.1470</td>\n",
       "      <td>0.14290</td>\n",
       "      <td>0.7788</td>\n",
       "      <td>0.1710</td>\n",
       "      <td>0.55690</td>\n",
       "      <td>1.089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110743781_A_G</td>\n",
       "      <td>110743781</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004329</td>\n",
       "      <td>0.010920</td>\n",
       "      <td>A</td>\n",
       "      <td>2.8420</td>\n",
       "      <td>0.09184</td>\n",
       "      <td>0.3939</td>\n",
       "      <td>0.5725</td>\n",
       "      <td>0.12830</td>\n",
       "      <td>1.210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110746300_C_A</td>\n",
       "      <td>110746300</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.7445</td>\n",
       "      <td>0.38820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110746313_C_T</td>\n",
       "      <td>110746313</td>\n",
       "      <td>T</td>\n",
       "      <td>0.008117</td>\n",
       "      <td>0.006550</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1163</td>\n",
       "      <td>0.73310</td>\n",
       "      <td>1.2410</td>\n",
       "      <td>0.6346</td>\n",
       "      <td>0.35780</td>\n",
       "      <td>4.305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110756712_G_A</td>\n",
       "      <td>110756712</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004870</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>0.6136</td>\n",
       "      <td>0.43340</td>\n",
       "      <td>2.2370</td>\n",
       "      <td>1.0550</td>\n",
       "      <td>0.28260</td>\n",
       "      <td>17.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110761239_G_A</td>\n",
       "      <td>110761239</td>\n",
       "      <td>A</td>\n",
       "      <td>0.067640</td>\n",
       "      <td>0.072050</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.73800</td>\n",
       "      <td>0.9343</td>\n",
       "      <td>0.2031</td>\n",
       "      <td>0.62760</td>\n",
       "      <td>1.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110761292_A_G</td>\n",
       "      <td>110761292</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002706</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.2420</td>\n",
       "      <td>0.26510</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5</td>\n",
       "      <td>chr5_110761460_C_CTTA</td>\n",
       "      <td>110761460</td>\n",
       "      <td>CTTA</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0.006550</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0300</td>\n",
       "      <td>0.31010</td>\n",
       "      <td>0.4940</td>\n",
       "      <td>0.7090</td>\n",
       "      <td>0.12310</td>\n",
       "      <td>1.983</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR                    SNP         BP    A1       F_A       F_U A2  \\\n",
       "0     5     chr5_110739266_C_T  110739266     T  0.015690  0.019650  C   \n",
       "1     5     chr5_110739267_C_A  110739267     A  0.000541  0.002183  C   \n",
       "2     5     chr5_110739354_G_A  110739354     A  0.003788  0.000000  G   \n",
       "3     5     chr5_110743749_T_C  110743749     C  0.087120  0.109200  T   \n",
       "4     5     chr5_110743781_A_G  110743781     G  0.004329  0.010920  A   \n",
       "5     5     chr5_110746300_C_A  110746300     A  0.001623  0.000000  C   \n",
       "6     5     chr5_110746313_C_T  110746313     T  0.008117  0.006550  C   \n",
       "7     5     chr5_110756712_G_A  110756712     A  0.004870  0.002183  G   \n",
       "8     5     chr5_110761239_G_A  110761239     A  0.067640  0.072050  G   \n",
       "9     5     chr5_110761292_A_G  110761292     G  0.002706  0.000000  A   \n",
       "10    5  chr5_110761460_C_CTTA  110761460  CTTA  0.003247  0.006550  C   \n",
       "\n",
       "     CHISQ        P      OR      SE      L95     U95  \n",
       "0   0.3548  0.55140  0.7954  0.3852  0.37380   1.692  \n",
       "1   1.1420  0.28520  0.2474  1.4150  0.01545   3.963  \n",
       "2   1.7400  0.18710     NaN     NaN      NaN     NaN  \n",
       "3   2.1470  0.14290  0.7788  0.1710  0.55690   1.089  \n",
       "4   2.8420  0.09184  0.3939  0.5725  0.12830   1.210  \n",
       "5   0.7445  0.38820     NaN     NaN      NaN     NaN  \n",
       "6   0.1163  0.73310  1.2410  0.6346  0.35780   4.305  \n",
       "7   0.6136  0.43340  2.2370  1.0550  0.28260  17.700  \n",
       "8   0.1119  0.73800  0.9343  0.2031  0.62760   1.391  \n",
       "9   1.2420  0.26510     NaN     NaN      NaN     NaN  \n",
       "10  1.0300  0.31010  0.4940  0.7090  0.12310   1.983  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding.assoc', delim_whitespace=True)\n",
    "#freq[freq['P']<0.05]\n",
    "freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.nonadj.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\r\n",
      "  --extract range SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all_coding.variantstoKeep\r\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.nonadj\r\n",
      "  --recode A\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "735 variants loaded from .bim file.\r\n",
      "1273 people (816 males, 457 females) loaded from .fam.\r\n",
      "1153 phenotype values loaded from .fam.\r\n",
      "--extract range: 724 variants excluded.\r\n",
      "--extract range: 11 variants remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 1273 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "11 variants and 1273 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 924 are cases and 229 are controls.  (120\r\n",
      "phenotypes are missing.)\r\n",
      "--recode A to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.nonadj.raw ...\r\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A).\n",
    "# Also extract the significant variants \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--extract range {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding.nonadj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding.nonadj.raw', delim_whitespace=True)\n",
    "recode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in all_ancestries for SLC25A46: 11\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr5_110739266_C_T_T',\n",
       " 'chr5_110739267_C_A_A',\n",
       " 'chr5_110739354_G_A_A',\n",
       " 'chr5_110743749_T_C_C',\n",
       " 'chr5_110743781_A_G_G',\n",
       " 'chr5_110746300_C_A_A',\n",
       " 'chr5_110746313_C_T_T',\n",
       " 'chr5_110756712_G_A_A',\n",
       " 'chr5_110761239_G_A_A',\n",
       " 'chr5_110761292_A_G_G',\n",
       " 'chr5_110761460_C_CTTA_CTTA']"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for SLC25A46: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_110739266_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>924</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>229</td>\n",
       "      <td>0.039301</td>\n",
       "      <td>chr5_110739266_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5_110739267_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110739267_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5_110739354_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>924</td>\n",
       "      <td>0.007576</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110739354_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5_110743749_T_C_C</td>\n",
       "      <td>10</td>\n",
       "      <td>141</td>\n",
       "      <td>924</td>\n",
       "      <td>0.163420</td>\n",
       "      <td>4</td>\n",
       "      <td>42</td>\n",
       "      <td>229</td>\n",
       "      <td>0.200873</td>\n",
       "      <td>chr5_110743749_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5_110743781_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>924</td>\n",
       "      <td>0.008658</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>229</td>\n",
       "      <td>0.021834</td>\n",
       "      <td>chr5_110743781_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5_110746300_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>924</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110746300_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5_110746313_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>924</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110746313_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5_110756712_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>924</td>\n",
       "      <td>0.009740</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110756712_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5_110761239_G_A_A</td>\n",
       "      <td>6</td>\n",
       "      <td>113</td>\n",
       "      <td>924</td>\n",
       "      <td>0.128788</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>229</td>\n",
       "      <td>0.144105</td>\n",
       "      <td>chr5_110761239_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr5_110761292_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>924</td>\n",
       "      <td>0.005411</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110761292_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr5_110761460_C_CTTA_CTTA</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>924</td>\n",
       "      <td>0.006494</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110761460_C_CTTA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0         chr5_110739266_C_T_T          1         27          924   \n",
       "1         chr5_110739267_C_A_A          0          1          924   \n",
       "2         chr5_110739354_G_A_A          0          7          924   \n",
       "3         chr5_110743749_T_C_C         10        141          924   \n",
       "4         chr5_110743781_A_G_G          0          8          924   \n",
       "5         chr5_110746300_C_A_A          0          3          924   \n",
       "6         chr5_110746313_C_T_T          1         13          924   \n",
       "7         chr5_110756712_G_A_A          0          9          924   \n",
       "8         chr5_110761239_G_A_A          6        113          924   \n",
       "9         chr5_110761292_A_G_G          0          5          924   \n",
       "10  chr5_110761460_C_CTTA_CTTA          0          6          924   \n",
       "\n",
       "    Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                0.030303             0             9             229   \n",
       "1                0.001082             0             1             229   \n",
       "2                0.007576             0             0             229   \n",
       "3                0.163420             4            42             229   \n",
       "4                0.008658             0             5             229   \n",
       "5                0.003247             0             0             229   \n",
       "6                0.015152             0             3             229   \n",
       "7                0.009740             0             1             229   \n",
       "8                0.128788             0            33             229   \n",
       "9                0.005411             0             0             229   \n",
       "10               0.006494             0             3             229   \n",
       "\n",
       "    Carrier freq in Controls                    SNP  \n",
       "0                   0.039301     chr5_110739266_C_T  \n",
       "1                   0.004367     chr5_110739267_C_A  \n",
       "2                   0.000000     chr5_110739354_G_A  \n",
       "3                   0.200873     chr5_110743749_T_C  \n",
       "4                   0.021834     chr5_110743781_A_G  \n",
       "5                   0.000000     chr5_110746300_C_A  \n",
       "6                   0.013100     chr5_110746313_C_T  \n",
       "7                   0.004367     chr5_110756712_G_A  \n",
       "8                   0.144105     chr5_110761239_G_A  \n",
       "9                   0.000000     chr5_110761292_A_G  \n",
       "10                  0.013100  chr5_110761460_C_CTTA  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['SNP'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>L95</th>\n",
       "      <th>OR</th>\n",
       "      <th>U95</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_110739266_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>924</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>229</td>\n",
       "      <td>0.039301</td>\n",
       "      <td>chr5_110739266_C_T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.015690</td>\n",
       "      <td>0.019650</td>\n",
       "      <td>C</td>\n",
       "      <td>0.37380</td>\n",
       "      <td>0.7954</td>\n",
       "      <td>1.692</td>\n",
       "      <td>0.55140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5_110739267_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110739267_C_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000541</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>C</td>\n",
       "      <td>0.01545</td>\n",
       "      <td>0.2474</td>\n",
       "      <td>3.963</td>\n",
       "      <td>0.28520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5_110739354_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>924</td>\n",
       "      <td>0.007576</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110739354_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003788</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.18710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5_110743749_T_C_C</td>\n",
       "      <td>10</td>\n",
       "      <td>141</td>\n",
       "      <td>924</td>\n",
       "      <td>0.163420</td>\n",
       "      <td>4</td>\n",
       "      <td>42</td>\n",
       "      <td>229</td>\n",
       "      <td>0.200873</td>\n",
       "      <td>chr5_110743749_T_C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.087120</td>\n",
       "      <td>0.109200</td>\n",
       "      <td>T</td>\n",
       "      <td>0.55690</td>\n",
       "      <td>0.7788</td>\n",
       "      <td>1.089</td>\n",
       "      <td>0.14290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5_110743781_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>924</td>\n",
       "      <td>0.008658</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>229</td>\n",
       "      <td>0.021834</td>\n",
       "      <td>chr5_110743781_A_G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004329</td>\n",
       "      <td>0.010920</td>\n",
       "      <td>A</td>\n",
       "      <td>0.12830</td>\n",
       "      <td>0.3939</td>\n",
       "      <td>1.210</td>\n",
       "      <td>0.09184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5_110746300_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>924</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110746300_C_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.38820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5_110746313_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>924</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110746313_C_T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.008117</td>\n",
       "      <td>0.006550</td>\n",
       "      <td>C</td>\n",
       "      <td>0.35780</td>\n",
       "      <td>1.2410</td>\n",
       "      <td>4.305</td>\n",
       "      <td>0.73310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5_110756712_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>924</td>\n",
       "      <td>0.009740</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110756712_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004870</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>G</td>\n",
       "      <td>0.28260</td>\n",
       "      <td>2.2370</td>\n",
       "      <td>17.700</td>\n",
       "      <td>0.43340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5_110761239_G_A_A</td>\n",
       "      <td>6</td>\n",
       "      <td>113</td>\n",
       "      <td>924</td>\n",
       "      <td>0.128788</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>229</td>\n",
       "      <td>0.144105</td>\n",
       "      <td>chr5_110761239_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.067640</td>\n",
       "      <td>0.072050</td>\n",
       "      <td>G</td>\n",
       "      <td>0.62760</td>\n",
       "      <td>0.9343</td>\n",
       "      <td>1.391</td>\n",
       "      <td>0.73800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr5_110761292_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>924</td>\n",
       "      <td>0.005411</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110761292_A_G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002706</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.26510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr5_110761460_C_CTTA_CTTA</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>924</td>\n",
       "      <td>0.006494</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110761460_C_CTTA</td>\n",
       "      <td>CTTA</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0.006550</td>\n",
       "      <td>C</td>\n",
       "      <td>0.12310</td>\n",
       "      <td>0.4940</td>\n",
       "      <td>1.983</td>\n",
       "      <td>0.31010</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0         chr5_110739266_C_T_T          1         27          924   \n",
       "1         chr5_110739267_C_A_A          0          1          924   \n",
       "2         chr5_110739354_G_A_A          0          7          924   \n",
       "3         chr5_110743749_T_C_C         10        141          924   \n",
       "4         chr5_110743781_A_G_G          0          8          924   \n",
       "5         chr5_110746300_C_A_A          0          3          924   \n",
       "6         chr5_110746313_C_T_T          1         13          924   \n",
       "7         chr5_110756712_G_A_A          0          9          924   \n",
       "8         chr5_110761239_G_A_A          6        113          924   \n",
       "9         chr5_110761292_A_G_G          0          5          924   \n",
       "10  chr5_110761460_C_CTTA_CTTA          0          6          924   \n",
       "\n",
       "    Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                0.030303             0             9             229   \n",
       "1                0.001082             0             1             229   \n",
       "2                0.007576             0             0             229   \n",
       "3                0.163420             4            42             229   \n",
       "4                0.008658             0             5             229   \n",
       "5                0.003247             0             0             229   \n",
       "6                0.015152             0             3             229   \n",
       "7                0.009740             0             1             229   \n",
       "8                0.128788             0            33             229   \n",
       "9                0.005411             0             0             229   \n",
       "10               0.006494             0             3             229   \n",
       "\n",
       "    Carrier freq in Controls                    SNP    A1       F_A       F_U  \\\n",
       "0                   0.039301     chr5_110739266_C_T     T  0.015690  0.019650   \n",
       "1                   0.004367     chr5_110739267_C_A     A  0.000541  0.002183   \n",
       "2                   0.000000     chr5_110739354_G_A     A  0.003788  0.000000   \n",
       "3                   0.200873     chr5_110743749_T_C     C  0.087120  0.109200   \n",
       "4                   0.021834     chr5_110743781_A_G     G  0.004329  0.010920   \n",
       "5                   0.000000     chr5_110746300_C_A     A  0.001623  0.000000   \n",
       "6                   0.013100     chr5_110746313_C_T     T  0.008117  0.006550   \n",
       "7                   0.004367     chr5_110756712_G_A     A  0.004870  0.002183   \n",
       "8                   0.144105     chr5_110761239_G_A     A  0.067640  0.072050   \n",
       "9                   0.000000     chr5_110761292_A_G     G  0.002706  0.000000   \n",
       "10                  0.013100  chr5_110761460_C_CTTA  CTTA  0.003247  0.006550   \n",
       "\n",
       "   A2      L95      OR     U95        P  \n",
       "0   C  0.37380  0.7954   1.692  0.55140  \n",
       "1   C  0.01545  0.2474   3.963  0.28520  \n",
       "2   G      NaN     NaN     NaN  0.18710  \n",
       "3   T  0.55690  0.7788   1.089  0.14290  \n",
       "4   A  0.12830  0.3939   1.210  0.09184  \n",
       "5   C      NaN     NaN     NaN  0.38820  \n",
       "6   C  0.35780  1.2410   4.305  0.73310  \n",
       "7   G  0.28260  2.2370  17.700  0.43340  \n",
       "8   G  0.62760  0.9343   1.391  0.73800  \n",
       "9   A      NaN     NaN     NaN  0.26510  \n",
       "10  C  0.12310  0.4940   1.983  0.31010  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge with the assoc file\n",
    "sig_merge = freq[['SNP','A1','F_A','F_U','A2','L95','OR','U95','P']]\n",
    "merged = pd.merge(df_results, sig_merge, on='SNP', how='right')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "## to CSV\n",
    "merged.to_csv(f'{WORK_DIR}/{ancestry_GROUP}_coding.variants-nonadj.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_coding.variants-nonadj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry_GROUP}_coding.variants-nonadj.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (4 May 2024)      www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\r\n",
      "  --covar SLC25A46_WGS_all_ancestries/EUR_covariate_file.txt\r\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\r\n",
      "  --covar-variance-standardize\r\n",
      "  --extract range SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all_coding.variantstoKeep\r\n",
      "  --glm\r\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding\r\n",
      "\r\n",
      "Start time: Tue May 21 21:09:33 2024\r\n",
      "3676 MiB RAM detected, ~2409 available; reserving 1838 MiB for main workspace.\r\n",
      "Using 1 compute thread.\r\n",
      "1273 samples (457 females, 816 males; 1273 founders) loaded from\r\n",
      "SLC25A46_WGS_all_ancestries/EUR_SLC25A46.fam.\r\n",
      "735 variants loaded from SLC25A46_WGS_all_ancestries/EUR_SLC25A46.bim.\r\n",
      "1 binary phenotype loaded (924 cases, 229 controls).\r\n",
      "--extract bed1: 724 variants excluded.\r\n",
      "7 covariates loaded from SLC25A46_WGS_all_ancestries/EUR_covariate_file.txt.\r\n",
      "--covar-variance-standardize: 7 covariates transformed.\r\n",
      "Calculating allele frequencies... 0%\b\b0%\b\bdone.\r\n",
      "11 variants remaining after main filters.\r\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\r\n",
      "phenotype 'PHENO1'.\r\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\b\b0%\b\bdone.\r\n",
      "Results written to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.PHENO1.glm.logistic.hybrid .\r\n",
      "End time: Tue May 21 21:09:33 2024\r\n"
     ]
    }
   ],
   "source": [
    "#No significant SNPs with/without AGE as covariate\n",
    "WORK_DIR = \"SLC25A46_WGS_all_ancestries\"\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--extract range {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--glm \\\n",
    "--covar {WORK_DIR}/{ancestry_GROUP}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>110739266</td>\n",
       "      <td>chr5_110739266_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.018031</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.348172</td>\n",
       "      <td>1.46798</td>\n",
       "      <td>-0.718714</td>\n",
       "      <td>0.472317</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>110739267</td>\n",
       "      <td>chr5_110739267_C_A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.188917</td>\n",
       "      <td>1.75190</td>\n",
       "      <td>-0.951222</td>\n",
       "      <td>0.341492</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5</td>\n",
       "      <td>110739354</td>\n",
       "      <td>chr5_110739354_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003467</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.101537</td>\n",
       "      <td>1.56963</td>\n",
       "      <td>-1.457240</td>\n",
       "      <td>0.145049</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>5</td>\n",
       "      <td>110743749</td>\n",
       "      <td>chr5_110743749_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Y</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.079750</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>1.369390</td>\n",
       "      <td>1.27448</td>\n",
       "      <td>0.246659</td>\n",
       "      <td>0.805172</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>5</td>\n",
       "      <td>110743781</td>\n",
       "      <td>chr5_110743781_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004854</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.140811</td>\n",
       "      <td>1.57818</td>\n",
       "      <td>-1.242150</td>\n",
       "      <td>0.214181</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>5</td>\n",
       "      <td>110746300</td>\n",
       "      <td>chr5_110746300_C_A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002080</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.007712</td>\n",
       "      <td>1.75738</td>\n",
       "      <td>-2.768300</td>\n",
       "      <td>0.005635</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>5</td>\n",
       "      <td>110746313</td>\n",
       "      <td>chr5_110746313_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.009015</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.292589</td>\n",
       "      <td>1.36368</td>\n",
       "      <td>-0.901225</td>\n",
       "      <td>0.367469</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>5</td>\n",
       "      <td>110756712</td>\n",
       "      <td>chr5_110756712_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.006241</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.087832</td>\n",
       "      <td>1.54080</td>\n",
       "      <td>-1.578620</td>\n",
       "      <td>0.114424</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>5</td>\n",
       "      <td>110761239</td>\n",
       "      <td>chr5_110761239_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.067961</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.920414</td>\n",
       "      <td>1.18426</td>\n",
       "      <td>-0.070028</td>\n",
       "      <td>0.944171</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>5</td>\n",
       "      <td>110761292</td>\n",
       "      <td>chr5_110761292_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000750</td>\n",
       "      <td>2.11995</td>\n",
       "      <td>-3.394350</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>5</td>\n",
       "      <td>110761460</td>\n",
       "      <td>chr5_110761460_C_CTTA</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTA</td>\n",
       "      <td>Y</td>\n",
       "      <td>CTTA</td>\n",
       "      <td>C</td>\n",
       "      <td>0.004161</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>721</td>\n",
       "      <td>0.021223</td>\n",
       "      <td>1.70564</td>\n",
       "      <td>-2.258790</td>\n",
       "      <td>0.023897</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM        POS                     ID REF   ALT PROVISIONAL_REF?    A1  \\\n",
       "0        5  110739266     chr5_110739266_C_T   C     T                Y     T   \n",
       "8        5  110739267     chr5_110739267_C_A   C     A                Y     A   \n",
       "16       5  110739354     chr5_110739354_G_A   G     A                Y     A   \n",
       "24       5  110743749     chr5_110743749_T_C   T     C                Y     C   \n",
       "32       5  110743781     chr5_110743781_A_G   A     G                Y     G   \n",
       "40       5  110746300     chr5_110746300_C_A   C     A                Y     A   \n",
       "48       5  110746313     chr5_110746313_C_T   C     T                Y     T   \n",
       "56       5  110756712     chr5_110756712_G_A   G     A                Y     A   \n",
       "64       5  110761239     chr5_110761239_G_A   G     A                Y     A   \n",
       "72       5  110761292     chr5_110761292_A_G   A     G                Y     G   \n",
       "80       5  110761460  chr5_110761460_C_CTTA   C  CTTA                Y  CTTA   \n",
       "\n",
       "   OMITTED   A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE    Z_STAT  \\\n",
       "0        C  0.018031      Y  ADD     721  0.348172     1.46798 -0.718714   \n",
       "8        C  0.000693      Y  ADD     721  0.188917     1.75190 -0.951222   \n",
       "16       G  0.003467      Y  ADD     721  0.101537     1.56963 -1.457240   \n",
       "24       T  0.079750      Y  ADD     721  1.369390     1.27448  0.246659   \n",
       "32       A  0.004854      Y  ADD     721  0.140811     1.57818 -1.242150   \n",
       "40       C  0.002080      Y  ADD     721  0.007712     1.75738 -2.768300   \n",
       "48       C  0.009015      Y  ADD     721  0.292589     1.36368 -0.901225   \n",
       "56       G  0.006241      Y  ADD     721  0.087832     1.54080 -1.578620   \n",
       "64       G  0.067961      Y  ADD     721  0.920414     1.18426 -0.070028   \n",
       "72       A  0.000693      Y  ADD     721  0.000750     2.11995 -3.394350   \n",
       "80       C  0.004161      Y  ADD     721  0.021223     1.70564 -2.258790   \n",
       "\n",
       "           P ERRCODE  \n",
       "0   0.472317       .  \n",
       "8   0.341492       .  \n",
       "16  0.145049       .  \n",
       "24  0.805172       .  \n",
       "32  0.214181       .  \n",
       "40  0.005635       .  \n",
       "48  0.367469       .  \n",
       "56  0.114424       .  \n",
       "64  0.944171       .  \n",
       "72  0.000688       .  \n",
       "80  0.023897       .  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "#assoc_add[assoc_add['P']<0.05]\n",
    "assoc_add"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile SLC25A46_WGS_all_ancestries/EUR_SLC25A46\r\n",
      "  --extract range SLC25A46_WGS_all_ancestries/EUR_SLC25A46.all_coding.variantstoKeep\r\n",
      "  --out SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding\r\n",
      "  --recode A\r\n",
      "\r\n",
      "3676 MB RAM detected; reserving 1838 MB for main workspace.\r\n",
      "735 variants loaded from .bim file.\r\n",
      "1273 people (816 males, 457 females) loaded from .fam.\r\n",
      "1153 phenotype values loaded from .fam.\r\n",
      "--extract range: 724 variants excluded.\r\n",
      "--extract range: 11 variants remaining.\r\n",
      "Using 1 thread.\r\n",
      "Before main variant filters, 1273 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "11 variants and 1273 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 924 are cases and 229 are controls.  (120\r\n",
      "phenotypes are missing.)\r\n",
      "--recode A to SLC25A46_WGS_all_ancestries/EUR_SLC25A46.coding.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "#--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A). \n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/{ancestry_GROUP}_SLC25A46 \\\n",
    "--extract range {WORK_DIR}/{ancestry_GROUP}_SLC25A46.all_coding.variantstoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry_GROUP}_SLC25A46.coding.raw', delim_whitespace=True)\n",
    "recode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in all_ancestries for SLC25A46: 11\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr5_110739266_C_T_T',\n",
       " 'chr5_110739267_C_A_A',\n",
       " 'chr5_110739354_G_A_A',\n",
       " 'chr5_110743749_T_C_C',\n",
       " 'chr5_110743781_A_G_G',\n",
       " 'chr5_110746300_C_A_A',\n",
       " 'chr5_110746313_C_T_T',\n",
       " 'chr5_110756712_G_A_A',\n",
       " 'chr5_110761239_G_A_A',\n",
       " 'chr5_110761292_A_G_G',\n",
       " 'chr5_110761460_C_CTTA_CTTA']"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for SLC25A46: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_110739266_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>924</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>229</td>\n",
       "      <td>0.039301</td>\n",
       "      <td>chr5_110739266_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5_110739267_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110739267_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5_110739354_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>924</td>\n",
       "      <td>0.007576</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110739354_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5_110743749_T_C_C</td>\n",
       "      <td>10</td>\n",
       "      <td>141</td>\n",
       "      <td>924</td>\n",
       "      <td>0.163420</td>\n",
       "      <td>4</td>\n",
       "      <td>42</td>\n",
       "      <td>229</td>\n",
       "      <td>0.200873</td>\n",
       "      <td>chr5_110743749_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5_110743781_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>924</td>\n",
       "      <td>0.008658</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>229</td>\n",
       "      <td>0.021834</td>\n",
       "      <td>chr5_110743781_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5_110746300_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>924</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110746300_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5_110746313_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>924</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110746313_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5_110756712_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>924</td>\n",
       "      <td>0.009740</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110756712_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5_110761239_G_A_A</td>\n",
       "      <td>6</td>\n",
       "      <td>113</td>\n",
       "      <td>924</td>\n",
       "      <td>0.128788</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>229</td>\n",
       "      <td>0.144105</td>\n",
       "      <td>chr5_110761239_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr5_110761292_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>924</td>\n",
       "      <td>0.005411</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110761292_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr5_110761460_C_CTTA_CTTA</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>924</td>\n",
       "      <td>0.006494</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110761460_C_CTTA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0         chr5_110739266_C_T_T          1         27          924   \n",
       "1         chr5_110739267_C_A_A          0          1          924   \n",
       "2         chr5_110739354_G_A_A          0          7          924   \n",
       "3         chr5_110743749_T_C_C         10        141          924   \n",
       "4         chr5_110743781_A_G_G          0          8          924   \n",
       "5         chr5_110746300_C_A_A          0          3          924   \n",
       "6         chr5_110746313_C_T_T          1         13          924   \n",
       "7         chr5_110756712_G_A_A          0          9          924   \n",
       "8         chr5_110761239_G_A_A          6        113          924   \n",
       "9         chr5_110761292_A_G_G          0          5          924   \n",
       "10  chr5_110761460_C_CTTA_CTTA          0          6          924   \n",
       "\n",
       "    Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                0.030303             0             9             229   \n",
       "1                0.001082             0             1             229   \n",
       "2                0.007576             0             0             229   \n",
       "3                0.163420             4            42             229   \n",
       "4                0.008658             0             5             229   \n",
       "5                0.003247             0             0             229   \n",
       "6                0.015152             0             3             229   \n",
       "7                0.009740             0             1             229   \n",
       "8                0.128788             0            33             229   \n",
       "9                0.005411             0             0             229   \n",
       "10               0.006494             0             3             229   \n",
       "\n",
       "    Carrier freq in Controls                     ID  \n",
       "0                   0.039301     chr5_110739266_C_T  \n",
       "1                   0.004367     chr5_110739267_C_A  \n",
       "2                   0.000000     chr5_110739354_G_A  \n",
       "3                   0.200873     chr5_110743749_T_C  \n",
       "4                   0.021834     chr5_110743781_A_G  \n",
       "5                   0.000000     chr5_110746300_C_A  \n",
       "6                   0.013100     chr5_110746313_C_T  \n",
       "7                   0.004367     chr5_110756712_G_A  \n",
       "8                   0.144105     chr5_110761239_G_A  \n",
       "9                   0.000000     chr5_110761292_A_G  \n",
       "10                  0.013100  chr5_110761460_C_CTTA  "
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['ID'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "      <th>A1</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_110739266_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>924</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>229</td>\n",
       "      <td>0.039301</td>\n",
       "      <td>chr5_110739266_C_T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.018031</td>\n",
       "      <td>721</td>\n",
       "      <td>0.348172</td>\n",
       "      <td>1.46798</td>\n",
       "      <td>-0.718714</td>\n",
       "      <td>0.472317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5_110739267_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>924</td>\n",
       "      <td>0.001082</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110739267_C_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.188917</td>\n",
       "      <td>1.75190</td>\n",
       "      <td>-0.951222</td>\n",
       "      <td>0.341492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5_110739354_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>924</td>\n",
       "      <td>0.007576</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110739354_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003467</td>\n",
       "      <td>721</td>\n",
       "      <td>0.101537</td>\n",
       "      <td>1.56963</td>\n",
       "      <td>-1.457240</td>\n",
       "      <td>0.145049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5_110743749_T_C_C</td>\n",
       "      <td>10</td>\n",
       "      <td>141</td>\n",
       "      <td>924</td>\n",
       "      <td>0.163420</td>\n",
       "      <td>4</td>\n",
       "      <td>42</td>\n",
       "      <td>229</td>\n",
       "      <td>0.200873</td>\n",
       "      <td>chr5_110743749_T_C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.079750</td>\n",
       "      <td>721</td>\n",
       "      <td>1.369390</td>\n",
       "      <td>1.27448</td>\n",
       "      <td>0.246659</td>\n",
       "      <td>0.805172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5_110743781_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>924</td>\n",
       "      <td>0.008658</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>229</td>\n",
       "      <td>0.021834</td>\n",
       "      <td>chr5_110743781_A_G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004854</td>\n",
       "      <td>721</td>\n",
       "      <td>0.140811</td>\n",
       "      <td>1.57818</td>\n",
       "      <td>-1.242150</td>\n",
       "      <td>0.214181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5_110746300_C_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>924</td>\n",
       "      <td>0.003247</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110746300_C_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002080</td>\n",
       "      <td>721</td>\n",
       "      <td>0.007712</td>\n",
       "      <td>1.75738</td>\n",
       "      <td>-2.768300</td>\n",
       "      <td>0.005635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr5_110746313_C_T_T</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>924</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110746313_C_T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.009015</td>\n",
       "      <td>721</td>\n",
       "      <td>0.292589</td>\n",
       "      <td>1.36368</td>\n",
       "      <td>-0.901225</td>\n",
       "      <td>0.367469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5_110756712_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>924</td>\n",
       "      <td>0.009740</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>0.004367</td>\n",
       "      <td>chr5_110756712_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.006241</td>\n",
       "      <td>721</td>\n",
       "      <td>0.087832</td>\n",
       "      <td>1.54080</td>\n",
       "      <td>-1.578620</td>\n",
       "      <td>0.114424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5_110761239_G_A_A</td>\n",
       "      <td>6</td>\n",
       "      <td>113</td>\n",
       "      <td>924</td>\n",
       "      <td>0.128788</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>229</td>\n",
       "      <td>0.144105</td>\n",
       "      <td>chr5_110761239_G_A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.067961</td>\n",
       "      <td>721</td>\n",
       "      <td>0.920414</td>\n",
       "      <td>1.18426</td>\n",
       "      <td>-0.070028</td>\n",
       "      <td>0.944171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr5_110761292_A_G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>924</td>\n",
       "      <td>0.005411</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>chr5_110761292_A_G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>721</td>\n",
       "      <td>0.000750</td>\n",
       "      <td>2.11995</td>\n",
       "      <td>-3.394350</td>\n",
       "      <td>0.000688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr5_110761460_C_CTTA_CTTA</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>924</td>\n",
       "      <td>0.006494</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>229</td>\n",
       "      <td>0.013100</td>\n",
       "      <td>chr5_110761460_C_CTTA</td>\n",
       "      <td>CTTA</td>\n",
       "      <td>0.004161</td>\n",
       "      <td>721</td>\n",
       "      <td>0.021223</td>\n",
       "      <td>1.70564</td>\n",
       "      <td>-2.258790</td>\n",
       "      <td>0.023897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Variant  Hom Cases  Het Cases  Total Cases  \\\n",
       "0         chr5_110739266_C_T_T          1         27          924   \n",
       "1         chr5_110739267_C_A_A          0          1          924   \n",
       "2         chr5_110739354_G_A_A          0          7          924   \n",
       "3         chr5_110743749_T_C_C         10        141          924   \n",
       "4         chr5_110743781_A_G_G          0          8          924   \n",
       "5         chr5_110746300_C_A_A          0          3          924   \n",
       "6         chr5_110746313_C_T_T          1         13          924   \n",
       "7         chr5_110756712_G_A_A          0          9          924   \n",
       "8         chr5_110761239_G_A_A          6        113          924   \n",
       "9         chr5_110761292_A_G_G          0          5          924   \n",
       "10  chr5_110761460_C_CTTA_CTTA          0          6          924   \n",
       "\n",
       "    Carrier freq in Cases  Hom Controls  Het Controls  Total Controls  \\\n",
       "0                0.030303             0             9             229   \n",
       "1                0.001082             0             1             229   \n",
       "2                0.007576             0             0             229   \n",
       "3                0.163420             4            42             229   \n",
       "4                0.008658             0             5             229   \n",
       "5                0.003247             0             0             229   \n",
       "6                0.015152             0             3             229   \n",
       "7                0.009740             0             1             229   \n",
       "8                0.128788             0            33             229   \n",
       "9                0.005411             0             0             229   \n",
       "10               0.006494             0             3             229   \n",
       "\n",
       "    Carrier freq in Controls                     ID    A1   A1_FREQ  OBS_CT  \\\n",
       "0                   0.039301     chr5_110739266_C_T     T  0.018031     721   \n",
       "1                   0.004367     chr5_110739267_C_A     A  0.000693     721   \n",
       "2                   0.000000     chr5_110739354_G_A     A  0.003467     721   \n",
       "3                   0.200873     chr5_110743749_T_C     C  0.079750     721   \n",
       "4                   0.021834     chr5_110743781_A_G     G  0.004854     721   \n",
       "5                   0.000000     chr5_110746300_C_A     A  0.002080     721   \n",
       "6                   0.013100     chr5_110746313_C_T     T  0.009015     721   \n",
       "7                   0.004367     chr5_110756712_G_A     A  0.006241     721   \n",
       "8                   0.144105     chr5_110761239_G_A     A  0.067961     721   \n",
       "9                   0.000000     chr5_110761292_A_G     G  0.000693     721   \n",
       "10                  0.013100  chr5_110761460_C_CTTA  CTTA  0.004161     721   \n",
       "\n",
       "          OR  LOG(OR)_SE    Z_STAT         P  \n",
       "0   0.348172     1.46798 -0.718714  0.472317  \n",
       "1   0.188917     1.75190 -0.951222  0.341492  \n",
       "2   0.101537     1.56963 -1.457240  0.145049  \n",
       "3   1.369390     1.27448  0.246659  0.805172  \n",
       "4   0.140811     1.57818 -1.242150  0.214181  \n",
       "5   0.007712     1.75738 -2.768300  0.005635  \n",
       "6   0.292589     1.36368 -0.901225  0.367469  \n",
       "7   0.087832     1.54080 -1.578620  0.114424  \n",
       "8   0.920414     1.18426 -0.070028  0.944171  \n",
       "9   0.000750     2.11995 -3.394350  0.000688  \n",
       "10  0.021223     1.70564 -2.258790  0.023897  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge with the glm file\n",
    "sig_merge = assoc_add[['ID','A1','A1_FREQ','OBS_CT','OR','LOG(OR)_SE','Z_STAT','P']]\n",
    "merged = pd.merge(df_results, sig_merge, on='ID', how='right')\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "## to CSV\n",
    "merged.to_csv(f'{WORK_DIR}/{ancestry_GROUP}_coding-variants-adj.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_coding-variants-adj.txt {WORKSPACE_BUCKET}/SLC25A46_{ancestry}/{ancestry_GROUP}_coding-variants-adj.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/SLC25A46_{ancestry_GROUP}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "165px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
